{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import GEOparse"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## miRNA (Multispecies Data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [],
   "source": [
    "GLP_PATH = 'GSE159033/GPL21572-124634.txt' # [miRNA-4] Affymetrix Multispecies miRNA-4 Array [ProbeSet ID version]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(GLP_PATH, sep='\\t', header=17, low_memory=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['ID', 'Accession_ID', 'Accession', 'Transcript ID(Array Design)',\n",
       "       'Sequence Type', 'Species Scientific Name', 'Alignments',\n",
       "       'Sequence Length', 'Sequence', 'Genome Context',\n",
       "       'Clustered miRNAs within 10kb', 'Target Genes', 'GeneChip Array',\n",
       "       'Annotation Date', 'Sequence Source', 'miRNA_ID', 'SPOT_ID'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Accession_ID</th>\n",
       "      <th>Accession</th>\n",
       "      <th>Transcript ID(Array Design)</th>\n",
       "      <th>Sequence Type</th>\n",
       "      <th>Species Scientific Name</th>\n",
       "      <th>Alignments</th>\n",
       "      <th>Sequence Length</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Genome Context</th>\n",
       "      <th>Clustered miRNAs within 10kb</th>\n",
       "      <th>Target Genes</th>\n",
       "      <th>GeneChip Array</th>\n",
       "      <th>Annotation Date</th>\n",
       "      <th>Sequence Source</th>\n",
       "      <th>miRNA_ID</th>\n",
       "      <th>SPOT_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20500000</td>\n",
       "      <td>MIMAT0000001_st</td>\n",
       "      <td>MIMAT0000001</td>\n",
       "      <td>cel-let-7-5p</td>\n",
       "      <td>miRNA</td>\n",
       "      <td>Caenorhabditis elegans</td>\n",
       "      <td>X:14744165-14744186 (-)</td>\n",
       "      <td>22</td>\n",
       "      <td>UGAGGUAGUAGGUUGUAUAGUU</td>\n",
       "      <td>C05G5.6 // sense // exon // 1</td>\n",
       "      <td>---</td>\n",
       "      <td>MTI // --- // mab-31 /// microcosm // AC7.2a.2...</td>\n",
       "      <td>miRNA-4_0</td>\n",
       "      <td>May 13, 2014</td>\n",
       "      <td>miRBase</td>\n",
       "      <td>cel-let-7-5p</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20500001</td>\n",
       "      <td>MIMAT0015091_st</td>\n",
       "      <td>MIMAT0015091</td>\n",
       "      <td>cel-let-7-3p</td>\n",
       "      <td>miRNA</td>\n",
       "      <td>Caenorhabditis elegans</td>\n",
       "      <td>X:14744123-14744147 (-)</td>\n",
       "      <td>25</td>\n",
       "      <td>UGAACUAUGCAAUUUUCUACCUUAC</td>\n",
       "      <td>C05G5.6 // sense // exon // 1</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>miRNA-4_0</td>\n",
       "      <td>May 13, 2014</td>\n",
       "      <td>miRBase</td>\n",
       "      <td>cel-let-7-3p</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         ID     Accession_ID     Accession Transcript ID(Array Design)  \\\n",
       "0  20500000  MIMAT0000001_st  MIMAT0000001                cel-let-7-5p   \n",
       "1  20500001  MIMAT0015091_st  MIMAT0015091                cel-let-7-3p   \n",
       "\n",
       "  Sequence Type Species Scientific Name               Alignments  \\\n",
       "0         miRNA  Caenorhabditis elegans  X:14744165-14744186 (-)   \n",
       "1         miRNA  Caenorhabditis elegans  X:14744123-14744147 (-)   \n",
       "\n",
       "  Sequence Length                   Sequence                 Genome Context  \\\n",
       "0              22     UGAGGUAGUAGGUUGUAUAGUU  C05G5.6 // sense // exon // 1   \n",
       "1              25  UGAACUAUGCAAUUUUCUACCUUAC  C05G5.6 // sense // exon // 1   \n",
       "\n",
       "  Clustered miRNAs within 10kb  \\\n",
       "0                          ---   \n",
       "1                          ---   \n",
       "\n",
       "                                        Target Genes GeneChip Array  \\\n",
       "0  MTI // --- // mab-31 /// microcosm // AC7.2a.2...      miRNA-4_0   \n",
       "1                                                ---      miRNA-4_0   \n",
       "\n",
       "  Annotation Date Sequence Source      miRNA_ID SPOT_ID  \n",
       "0    May 13, 2014         miRBase  cel-let-7-5p     NaN  \n",
       "1    May 13, 2014         miRBase  cel-let-7-3p     NaN  "
      ]
     },
     "execution_count": 89,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Caenorhabditis elegans', 'Homo sapiens',\n",
       "       'Drosophila melanogaster', 'Mus musculus', 'Arabidopsis thaliana',\n",
       "       'Caenorhabditis briggsae', 'Rattus norvegicus', 'Oryza sativa',\n",
       "       'Epstein Barr virus', 'Gallus gallus', 'Drosophila pseudoobscura',\n",
       "       'Danio rerio', 'Xenopus laevis', 'Zea mays', 'Sorghum bicolor',\n",
       "       'Ovis aries', 'Apis mellifera', 'Anopheles gambiae',\n",
       "       'Canis familiaris', 'Mouse gammaherpesvirus 68',\n",
       "       'Human cytomegalovirus', 'Medicago truncatula',\n",
       "       'Saccharum officinarum', 'Glycine max', 'Populus trichocarpa',\n",
       "       'Sus scrofa', 'Kaposi sarcoma-associated herpesvirus',\n",
       "       'Macaca mulatta', 'Gorilla gorilla', 'Pongo pygmaeus',\n",
       "       'Pan paniscus', 'Ateles geoffroyi', 'Pan troglodytes',\n",
       "       'Lagothrix lagotricha', 'Macaca nemestrina', 'Saguinus labiatus',\n",
       "       'Lemur catta', 'Fugu rubripes', 'Tetraodon nigroviridis',\n",
       "       'Physcomitrella patens', 'Simian virus 40',\n",
       "       'Rhesus lymphocryptovirus', 'Herpes Simplex Virus 1', 'Bos taurus',\n",
       "       'Xenopus tropicalis', 'Bombyx mori', 'Mareks disease virus',\n",
       "       'Schmidtea mediterranea', 'Monodelphis domestica',\n",
       "       'Gossypium hirsutum', 'Gossypium raimondii', 'Gossypium herbaceum',\n",
       "       'Chlamydomonas reinhardtii', 'Rhesus monkey rhadinovirus',\n",
       "       'Cricetulus griseus', 'Brassica napus', 'Pinus taeda',\n",
       "       'Mareks disease virus type 2', 'Selaginella moellendorffii',\n",
       "       'Human immunodeficiency virus 1', 'Triticum aestivum',\n",
       "       'Dictyostelium discoideum', 'Mouse cytomegalovirus',\n",
       "       'Carica papaya', 'Brassica oleracea', 'Brassica rapa',\n",
       "       'Vitis vinifera', 'Symphalangus syndactylus', 'Pygathrix bieti',\n",
       "       'Ornithorhynchus anatinus', 'Oikopleura dioica',\n",
       "       'Ciona intestinalis', 'Ciona savignyi', 'Solanum lycopersicum',\n",
       "       'Tribolium castaneum', 'Drosophila ananassae', 'Drosophila erecta',\n",
       "       'Drosophila grimshawi', 'Drosophila mojavensis',\n",
       "       'Drosophila persimilis', 'Drosophila sechellia',\n",
       "       'Drosophila simulans', 'Drosophila virilis',\n",
       "       'Drosophila willistoni', 'Drosophila yakuba', 'JC polyomavirus',\n",
       "       'BK polyomavirus', 'Branchiostoma floridae', 'Capitella teleta',\n",
       "       'Lottia gigantea', 'Nematostella vectensis',\n",
       "       'Saccoglossus kowalevskii', 'Strongylocentrotus purpuratus',\n",
       "       'Amphimedon queenslandica', 'Cerebratulus lacteus',\n",
       "       'Haliotis rufescens', 'Hydra magnipapillata', 'Ixodes scapularis',\n",
       "       'Lotus japonicus', 'Vigna unguiculata', 'Locusta migratoria',\n",
       "       'Merkel cell polyomavirus', 'Schistosoma japonicum',\n",
       "       'Schistosoma mansoni', 'Herpes Simplex Virus 2', 'Malus domestica',\n",
       "       'Phaseolus vulgaris', 'Caenorhabditis remanei',\n",
       "       'Pristionchus pacificus', 'Herpes B virus',\n",
       "       'Brachypodium distachyon', 'Aquilegia caerulea', 'Daphnia pulex',\n",
       "       'Herpesvirus of turkeys', 'Infectious laryngotracheitis virus',\n",
       "       'Populus euphratica', 'Equus caballus', 'Oryzias latipes',\n",
       "       'Bovine herpesvirus 1', 'Citrus trifoliata', 'Citrus sinensis',\n",
       "       'Citrus clementine', 'Citrus reticulata', 'Brugia malayi',\n",
       "       'Acyrthosiphon pisum', 'Ricinus communis', 'Aedes aegypti',\n",
       "       'Gossypium arboreum', 'Culex quinquefasciatus',\n",
       "       'Taeniopygia guttata', 'Arabidopsis lyrata',\n",
       "       'Ectocarpus siliculosus', 'Nasonia vitripennis',\n",
       "       'Arachis hypogaea', 'Glycine soja', 'Picea abies',\n",
       "       'Strigamia maritima', 'Hordeum vulgare', 'Triticum turgidum',\n",
       "       'Aegilops tauschii', 'Festuca arundinacea', 'Nasonia giraulti',\n",
       "       'Nasonia longicornis', 'Bruguiera gymnorhiza',\n",
       "       'Bruguiera cylindrica', 'Petromyzon marinus', 'Theobroma cacao',\n",
       "       'Xenoturbella bocki', 'Echinococcus granulosus',\n",
       "       'Echinococcus multilocularis', 'Heliconius melpomene',\n",
       "       'Glottidia pyramidata', 'Terebratulina retusa',\n",
       "       'Rehmannia glutinosa', 'Cucumis melo',\n",
       "       'Bandicoot papillomatosis carcinomatosis virus type 1',\n",
       "       'Bandicoot papillomatosis carcinomatosis virus type 2',\n",
       "       'Saccharum ssp.', 'Rhipicephalus microplus', 'Ascaris suum',\n",
       "       'Anolis carolinensis', 'Acacia auriculiformis', 'Acacia mangium',\n",
       "       'Phaeodactylum tricornutum', 'Herpesvirus saimiri strain A11',\n",
       "       'Macropus eugenii', 'Salvia sclarea', 'Tetranychus urticae',\n",
       "       'Sarcophilus harrisii', 'Haemonchus contortus',\n",
       "       'Human herpesvirus 6B', 'Digitalis purpurea', 'Nicotiana tabacum',\n",
       "       'Solanum tuberosum', 'Elaeis guineensis', 'Manihot esculenta',\n",
       "       'Manduca sexta', 'Cynara cardunculus', 'Linum usitatissimum',\n",
       "       'Panax ginseng', 'Hevea brasiliensis', 'Prunus persica',\n",
       "       'Marsupenaeus japonicus', 'Pseudorabies virus', 'Pinus densata',\n",
       "       'Paralichthys olivaceus', 'Helianthus annuus',\n",
       "       'Helianthus ciliaris', 'Helianthus tuberosus',\n",
       "       'Helianthus argophyllus', 'Helianthus paradoxus',\n",
       "       'Helianthus petiolaris', 'Helianthus exilis',\n",
       "       'Hippoglossus hippoglossus', 'Bovine leukemia virus',\n",
       "       'Cunninghamia lanceolata', 'Cyprinus carpio',\n",
       "       'Duck enteritis virus', 'Artibeus jamaicensis',\n",
       "       'Ictalurus punctatus', 'Caenorhabditis brenneri', 'Sycon ciliatum',\n",
       "       'Leucosolenia complicata', 'Avicennia marina',\n",
       "       'Panagrellus redivivus', 'Branchiostoma belcheri', '---'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['Species Scientific Name'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [],
   "source": [
    "miRNA_homo_sapiens_df = df[df['Species Scientific Name']=='Homo sapiens']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Accession_ID</th>\n",
       "      <th>Accession</th>\n",
       "      <th>Transcript ID(Array Design)</th>\n",
       "      <th>Sequence Type</th>\n",
       "      <th>Species Scientific Name</th>\n",
       "      <th>Alignments</th>\n",
       "      <th>Sequence Length</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Genome Context</th>\n",
       "      <th>Clustered miRNAs within 10kb</th>\n",
       "      <th>Target Genes</th>\n",
       "      <th>GeneChip Array</th>\n",
       "      <th>Annotation Date</th>\n",
       "      <th>Sequence Source</th>\n",
       "      <th>miRNA_ID</th>\n",
       "      <th>SPOT_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>20500112</td>\n",
       "      <td>MIMAT0000062_st</td>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>miRNA</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>chr11:122017276-122017297 (-) /// chr22:465086...</td>\n",
       "      <td>22</td>\n",
       "      <td>UGAGGUAGUAGGUUGUAUAGUU</td>\n",
       "      <td>ENST00000362295 // sense // exon // 1 /// ENST...</td>\n",
       "      <td>hsa-let-7a-3 // chr22:46508629-46508702 (+) //...</td>\n",
       "      <td>MTI // --- // CCNG1 /// microcosm // ENST00000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113</th>\n",
       "      <td>20500113</td>\n",
       "      <td>MIMAT0004481_st</td>\n",
       "      <td>MIMAT0004481</td>\n",
       "      <td>hsa-let-7a-3p</td>\n",
       "      <td>miRNA</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>chr22:46508680-46508700 (+) /// chr9:96938295-...</td>\n",
       "      <td>21</td>\n",
       "      <td>CUAUACAAUCUACUGUCUUUC</td>\n",
       "      <td>ENST00000362295 // sense // exon // 1 /// ENST...</td>\n",
       "      <td>hsa-let-7a-3 // chr22:46508629-46508702 (+) //...</td>\n",
       "      <td>MTI // --- // SENP7 /// microcosm // ENST00000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>hsa-let-7a-3p</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>20500114</td>\n",
       "      <td>MIMAT0010195_st</td>\n",
       "      <td>MIMAT0010195</td>\n",
       "      <td>hsa-let-7a-2-3p</td>\n",
       "      <td>miRNA</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>chr11:122017231-122017252 (-)</td>\n",
       "      <td>22</td>\n",
       "      <td>CUGUACAGCCUCCUAGCUUUCC</td>\n",
       "      <td>ENST00000362105 // sense // exon // 1</td>\n",
       "      <td>hsa-let-7a-2 // chr11:122017230-122017301 (-) ...</td>\n",
       "      <td>---</td>\n",
       "      <td>miRNA-4_0</td>\n",
       "      <td>May 13, 2014</td>\n",
       "      <td>miRBase</td>\n",
       "      <td>hsa-let-7a-2-3p</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>20500115</td>\n",
       "      <td>MIMAT0000063_st</td>\n",
       "      <td>MIMAT0000063</td>\n",
       "      <td>hsa-let-7b-5p</td>\n",
       "      <td>miRNA</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>chr22:46509571-46509592 (+)</td>\n",
       "      <td>22</td>\n",
       "      <td>UGAGGUAGUAGGUUGUGUGGUU</td>\n",
       "      <td>ENST00000360737 // sense // exon // 5 /// OTTH...</td>\n",
       "      <td>hsa-let-7a-3 // chr22:46508629-46508702 (+) //...</td>\n",
       "      <td>MTI // --- // WNK1 /// microcosm // ENST000000...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>hsa-let-7b-5p</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116</th>\n",
       "      <td>20500116</td>\n",
       "      <td>MIMAT0004482_st</td>\n",
       "      <td>MIMAT0004482</td>\n",
       "      <td>hsa-let-7b-3p</td>\n",
       "      <td>miRNA</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>chr22:46509625-46509646 (+)</td>\n",
       "      <td>22</td>\n",
       "      <td>CUAUACAACCUACUGCCUUCCC</td>\n",
       "      <td>ENST00000360737 // sense // exon // 5 /// OTTH...</td>\n",
       "      <td>hsa-let-7a-3 // chr22:46508629-46508702 (+) //...</td>\n",
       "      <td>MTI // --- // KIAA1715 /// microcosm // ENST00...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>hsa-let-7b-3p</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36348</th>\n",
       "      <td>20538487</td>\n",
       "      <td>gi555853_copy5_st</td>\n",
       "      <td>gi555853_copy5</td>\n",
       "      <td>gi:555853</td>\n",
       "      <td>5.8s rRNA</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>---</td>\n",
       "      <td>157</td>\n",
       "      <td>CGACTCTTAGCGGTGGATCACTCGGCTCGTGCGTCGATGAAGAACG...</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>miRNA-4_0</td>\n",
       "      <td>May 13, 2014</td>\n",
       "      <td>Affymetrix Proprietary Database</td>\n",
       "      <td>NaN</td>\n",
       "      <td>--5.8s rRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36349</th>\n",
       "      <td>20538488</td>\n",
       "      <td>gi555853_copy6_st</td>\n",
       "      <td>gi555853_copy6</td>\n",
       "      <td>gi:555853</td>\n",
       "      <td>5.8s rRNA</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>---</td>\n",
       "      <td>157</td>\n",
       "      <td>CGACTCTTAGCGGTGGATCACTCGGCTCGTGCGTCGATGAAGAACG...</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>miRNA-4_0</td>\n",
       "      <td>May 13, 2014</td>\n",
       "      <td>Affymetrix Proprietary Database</td>\n",
       "      <td>NaN</td>\n",
       "      <td>--5.8s rRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36350</th>\n",
       "      <td>20538489</td>\n",
       "      <td>gi555853_copy7_st</td>\n",
       "      <td>gi555853_copy7</td>\n",
       "      <td>gi:555853</td>\n",
       "      <td>5.8s rRNA</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>---</td>\n",
       "      <td>157</td>\n",
       "      <td>CGACTCTTAGCGGTGGATCACTCGGCTCGTGCGTCGATGAAGAACG...</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>miRNA-4_0</td>\n",
       "      <td>May 13, 2014</td>\n",
       "      <td>Affymetrix Proprietary Database</td>\n",
       "      <td>NaN</td>\n",
       "      <td>--5.8s rRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36351</th>\n",
       "      <td>20538490</td>\n",
       "      <td>gi555853_copy8_st</td>\n",
       "      <td>gi555853_copy8</td>\n",
       "      <td>gi:555853</td>\n",
       "      <td>5.8s rRNA</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>---</td>\n",
       "      <td>157</td>\n",
       "      <td>CGACTCTTAGCGGTGGATCACTCGGCTCGTGCGTCGATGAAGAACG...</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>miRNA-4_0</td>\n",
       "      <td>May 13, 2014</td>\n",
       "      <td>Affymetrix Proprietary Database</td>\n",
       "      <td>NaN</td>\n",
       "      <td>--5.8s rRNA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36352</th>\n",
       "      <td>20538491</td>\n",
       "      <td>gi555853_copy9_st</td>\n",
       "      <td>gi555853_copy9</td>\n",
       "      <td>gi:555853</td>\n",
       "      <td>5.8s rRNA</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>---</td>\n",
       "      <td>157</td>\n",
       "      <td>CGACTCTTAGCGGTGGATCACTCGGCTCGTGCGTCGATGAAGAACG...</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>---</td>\n",
       "      <td>miRNA-4_0</td>\n",
       "      <td>May 13, 2014</td>\n",
       "      <td>Affymetrix Proprietary Database</td>\n",
       "      <td>NaN</td>\n",
       "      <td>--5.8s rRNA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6631 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             ID       Accession_ID       Accession  \\\n",
       "112    20500112    MIMAT0000062_st    MIMAT0000062   \n",
       "113    20500113    MIMAT0004481_st    MIMAT0004481   \n",
       "114    20500114    MIMAT0010195_st    MIMAT0010195   \n",
       "115    20500115    MIMAT0000063_st    MIMAT0000063   \n",
       "116    20500116    MIMAT0004482_st    MIMAT0004482   \n",
       "...         ...                ...             ...   \n",
       "36348  20538487  gi555853_copy5_st  gi555853_copy5   \n",
       "36349  20538488  gi555853_copy6_st  gi555853_copy6   \n",
       "36350  20538489  gi555853_copy7_st  gi555853_copy7   \n",
       "36351  20538490  gi555853_copy8_st  gi555853_copy8   \n",
       "36352  20538491  gi555853_copy9_st  gi555853_copy9   \n",
       "\n",
       "      Transcript ID(Array Design) Sequence Type Species Scientific Name  \\\n",
       "112                 hsa-let-7a-5p         miRNA            Homo sapiens   \n",
       "113                 hsa-let-7a-3p         miRNA            Homo sapiens   \n",
       "114               hsa-let-7a-2-3p         miRNA            Homo sapiens   \n",
       "115                 hsa-let-7b-5p         miRNA            Homo sapiens   \n",
       "116                 hsa-let-7b-3p         miRNA            Homo sapiens   \n",
       "...                           ...           ...                     ...   \n",
       "36348                   gi:555853     5.8s rRNA            Homo sapiens   \n",
       "36349                   gi:555853     5.8s rRNA            Homo sapiens   \n",
       "36350                   gi:555853     5.8s rRNA            Homo sapiens   \n",
       "36351                   gi:555853     5.8s rRNA            Homo sapiens   \n",
       "36352                   gi:555853     5.8s rRNA            Homo sapiens   \n",
       "\n",
       "                                              Alignments Sequence Length  \\\n",
       "112    chr11:122017276-122017297 (-) /// chr22:465086...              22   \n",
       "113    chr22:46508680-46508700 (+) /// chr9:96938295-...              21   \n",
       "114                        chr11:122017231-122017252 (-)              22   \n",
       "115                          chr22:46509571-46509592 (+)              22   \n",
       "116                          chr22:46509625-46509646 (+)              22   \n",
       "...                                                  ...             ...   \n",
       "36348                                                ---             157   \n",
       "36349                                                ---             157   \n",
       "36350                                                ---             157   \n",
       "36351                                                ---             157   \n",
       "36352                                                ---             157   \n",
       "\n",
       "                                                Sequence  \\\n",
       "112                               UGAGGUAGUAGGUUGUAUAGUU   \n",
       "113                                CUAUACAAUCUACUGUCUUUC   \n",
       "114                               CUGUACAGCCUCCUAGCUUUCC   \n",
       "115                               UGAGGUAGUAGGUUGUGUGGUU   \n",
       "116                               CUAUACAACCUACUGCCUUCCC   \n",
       "...                                                  ...   \n",
       "36348  CGACTCTTAGCGGTGGATCACTCGGCTCGTGCGTCGATGAAGAACG...   \n",
       "36349  CGACTCTTAGCGGTGGATCACTCGGCTCGTGCGTCGATGAAGAACG...   \n",
       "36350  CGACTCTTAGCGGTGGATCACTCGGCTCGTGCGTCGATGAAGAACG...   \n",
       "36351  CGACTCTTAGCGGTGGATCACTCGGCTCGTGCGTCGATGAAGAACG...   \n",
       "36352  CGACTCTTAGCGGTGGATCACTCGGCTCGTGCGTCGATGAAGAACG...   \n",
       "\n",
       "                                          Genome Context  \\\n",
       "112    ENST00000362295 // sense // exon // 1 /// ENST...   \n",
       "113    ENST00000362295 // sense // exon // 1 /// ENST...   \n",
       "114                ENST00000362105 // sense // exon // 1   \n",
       "115    ENST00000360737 // sense // exon // 5 /// OTTH...   \n",
       "116    ENST00000360737 // sense // exon // 5 /// OTTH...   \n",
       "...                                                  ...   \n",
       "36348                                                ---   \n",
       "36349                                                ---   \n",
       "36350                                                ---   \n",
       "36351                                                ---   \n",
       "36352                                                ---   \n",
       "\n",
       "                            Clustered miRNAs within 10kb  \\\n",
       "112    hsa-let-7a-3 // chr22:46508629-46508702 (+) //...   \n",
       "113    hsa-let-7a-3 // chr22:46508629-46508702 (+) //...   \n",
       "114    hsa-let-7a-2 // chr11:122017230-122017301 (-) ...   \n",
       "115    hsa-let-7a-3 // chr22:46508629-46508702 (+) //...   \n",
       "116    hsa-let-7a-3 // chr22:46508629-46508702 (+) //...   \n",
       "...                                                  ...   \n",
       "36348                                                ---   \n",
       "36349                                                ---   \n",
       "36350                                                ---   \n",
       "36351                                                ---   \n",
       "36352                                                ---   \n",
       "\n",
       "                                            Target Genes GeneChip Array  \\\n",
       "112    MTI // --- // CCNG1 /// microcosm // ENST00000...            NaN   \n",
       "113    MTI // --- // SENP7 /// microcosm // ENST00000...            NaN   \n",
       "114                                                  ---      miRNA-4_0   \n",
       "115    MTI // --- // WNK1 /// microcosm // ENST000000...            NaN   \n",
       "116    MTI // --- // KIAA1715 /// microcosm // ENST00...            NaN   \n",
       "...                                                  ...            ...   \n",
       "36348                                                ---      miRNA-4_0   \n",
       "36349                                                ---      miRNA-4_0   \n",
       "36350                                                ---      miRNA-4_0   \n",
       "36351                                                ---      miRNA-4_0   \n",
       "36352                                                ---      miRNA-4_0   \n",
       "\n",
       "      Annotation Date                  Sequence Source         miRNA_ID  \\\n",
       "112               NaN                              NaN    hsa-let-7a-5p   \n",
       "113               NaN                              NaN    hsa-let-7a-3p   \n",
       "114      May 13, 2014                          miRBase  hsa-let-7a-2-3p   \n",
       "115               NaN                              NaN    hsa-let-7b-5p   \n",
       "116               NaN                              NaN    hsa-let-7b-3p   \n",
       "...               ...                              ...              ...   \n",
       "36348    May 13, 2014  Affymetrix Proprietary Database              NaN   \n",
       "36349    May 13, 2014  Affymetrix Proprietary Database              NaN   \n",
       "36350    May 13, 2014  Affymetrix Proprietary Database              NaN   \n",
       "36351    May 13, 2014  Affymetrix Proprietary Database              NaN   \n",
       "36352    May 13, 2014  Affymetrix Proprietary Database              NaN   \n",
       "\n",
       "           SPOT_ID  \n",
       "112            NaN  \n",
       "113            NaN  \n",
       "114            NaN  \n",
       "115            NaN  \n",
       "116            NaN  \n",
       "...            ...  \n",
       "36348  --5.8s rRNA  \n",
       "36349  --5.8s rRNA  \n",
       "36350  --5.8s rRNA  \n",
       "36351  --5.8s rRNA  \n",
       "36352  --5.8s rRNA  \n",
       "\n",
       "[6631 rows x 17 columns]"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "miRNA_homo_sapiens_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## SOFT Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [],
   "source": [
    "FAMILY_PATH = 'GSE159033/GSE159033_family.soft\\GSE159033_family.soft'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "10-Apr-2024 18:30:28 INFO GEOparse - Parsing GSE159033/GSE159033_family.soft\\GSE159033_family.soft: \n",
      "10-Apr-2024 18:30:28 DEBUG GEOparse - DATABASE: GeoMiame\n",
      "10-Apr-2024 18:30:28 DEBUG GEOparse - SERIES: GSE159033\n",
      "10-Apr-2024 18:30:28 DEBUG GEOparse - PLATFORM: GPL21572\n",
      "c:\\Users\\giuli\\anaconda3\\lib\\site-packages\\GEOparse\\GEOparse.py:401: DtypeWarning: Columns (7) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  return read_csv(StringIO(data), index_col=None, sep=\"\\t\")\n",
      "10-Apr-2024 18:30:30 DEBUG GEOparse - SAMPLE: GSM4817989\n",
      "10-Apr-2024 18:30:30 DEBUG GEOparse - SAMPLE: GSM4817990\n",
      "10-Apr-2024 18:30:30 DEBUG GEOparse - SAMPLE: GSM4817991\n",
      "10-Apr-2024 18:30:30 DEBUG GEOparse - SAMPLE: GSM4817992\n",
      "10-Apr-2024 18:30:30 DEBUG GEOparse - SAMPLE: GSM4817993\n",
      "10-Apr-2024 18:30:30 DEBUG GEOparse - SAMPLE: GSM4817994\n",
      "10-Apr-2024 18:30:30 DEBUG GEOparse - SAMPLE: GSM4817995\n",
      "10-Apr-2024 18:30:30 DEBUG GEOparse - SAMPLE: GSM4817996\n",
      "10-Apr-2024 18:30:30 DEBUG GEOparse - SAMPLE: GSM4817997\n",
      "10-Apr-2024 18:30:30 DEBUG GEOparse - SAMPLE: GSM4817998\n",
      "10-Apr-2024 18:30:30 DEBUG GEOparse - SAMPLE: GSM4817999\n"
     ]
    }
   ],
   "source": [
    "gse_family = GEOparse.get_GEO(filepath=FAMILY_PATH)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<SERIES: GSE159033 - 11 SAMPLES, 1 d(s)>"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gse_family"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>geo_accession</th>\n",
       "      <th>status</th>\n",
       "      <th>submission_date</th>\n",
       "      <th>last_update_date</th>\n",
       "      <th>type</th>\n",
       "      <th>channel_count</th>\n",
       "      <th>source_name_ch1</th>\n",
       "      <th>organism_ch1</th>\n",
       "      <th>taxid_ch1</th>\n",
       "      <th>...</th>\n",
       "      <th>contact_name</th>\n",
       "      <th>contact_email</th>\n",
       "      <th>contact_institute</th>\n",
       "      <th>contact_address</th>\n",
       "      <th>contact_city</th>\n",
       "      <th>contact_zip/postal_code</th>\n",
       "      <th>contact_country</th>\n",
       "      <th>supplementary_file</th>\n",
       "      <th>series_id</th>\n",
       "      <th>data_row_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GSM4817989</th>\n",
       "      <td>Sample04-positive</td>\n",
       "      <td>GSM4817989</td>\n",
       "      <td>Public on Dec 21 2020</td>\n",
       "      <td>Oct 05 2020</td>\n",
       "      <td>Dec 22 2020</td>\n",
       "      <td>RNA</td>\n",
       "      <td>1</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>9606</td>\n",
       "      <td>...</td>\n",
       "      <td>Maider,,MuÃ±oz-Culla</td>\n",
       "      <td>maider.munoz@ehu.eus</td>\n",
       "      <td>Biodonostia Institute</td>\n",
       "      <td>Paseo Dr.Begiristain s/n</td>\n",
       "      <td>San Sebastian</td>\n",
       "      <td>20014</td>\n",
       "      <td>Spain</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4817...</td>\n",
       "      <td>GSE159033,GSE159036</td>\n",
       "      <td>6631</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM4817990</th>\n",
       "      <td>Sample05-positive</td>\n",
       "      <td>GSM4817990</td>\n",
       "      <td>Public on Dec 21 2020</td>\n",
       "      <td>Oct 05 2020</td>\n",
       "      <td>Dec 22 2020</td>\n",
       "      <td>RNA</td>\n",
       "      <td>1</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>9606</td>\n",
       "      <td>...</td>\n",
       "      <td>Maider,,MuÃ±oz-Culla</td>\n",
       "      <td>maider.munoz@ehu.eus</td>\n",
       "      <td>Biodonostia Institute</td>\n",
       "      <td>Paseo Dr.Begiristain s/n</td>\n",
       "      <td>San Sebastian</td>\n",
       "      <td>20014</td>\n",
       "      <td>Spain</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4817...</td>\n",
       "      <td>GSE159033,GSE159036</td>\n",
       "      <td>6631</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM4817991</th>\n",
       "      <td>Sample11-positive</td>\n",
       "      <td>GSM4817991</td>\n",
       "      <td>Public on Dec 21 2020</td>\n",
       "      <td>Oct 05 2020</td>\n",
       "      <td>Dec 22 2020</td>\n",
       "      <td>RNA</td>\n",
       "      <td>1</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>9606</td>\n",
       "      <td>...</td>\n",
       "      <td>Maider,,MuÃ±oz-Culla</td>\n",
       "      <td>maider.munoz@ehu.eus</td>\n",
       "      <td>Biodonostia Institute</td>\n",
       "      <td>Paseo Dr.Begiristain s/n</td>\n",
       "      <td>San Sebastian</td>\n",
       "      <td>20014</td>\n",
       "      <td>Spain</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4817...</td>\n",
       "      <td>GSE159033,GSE159036</td>\n",
       "      <td>6631</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM4817992</th>\n",
       "      <td>Sample14-positive</td>\n",
       "      <td>GSM4817992</td>\n",
       "      <td>Public on Dec 21 2020</td>\n",
       "      <td>Oct 05 2020</td>\n",
       "      <td>Dec 22 2020</td>\n",
       "      <td>RNA</td>\n",
       "      <td>1</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>9606</td>\n",
       "      <td>...</td>\n",
       "      <td>Maider,,MuÃ±oz-Culla</td>\n",
       "      <td>maider.munoz@ehu.eus</td>\n",
       "      <td>Biodonostia Institute</td>\n",
       "      <td>Paseo Dr.Begiristain s/n</td>\n",
       "      <td>San Sebastian</td>\n",
       "      <td>20014</td>\n",
       "      <td>Spain</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4817...</td>\n",
       "      <td>GSE159033,GSE159036</td>\n",
       "      <td>6631</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM4817993</th>\n",
       "      <td>Sample19-positive</td>\n",
       "      <td>GSM4817993</td>\n",
       "      <td>Public on Dec 21 2020</td>\n",
       "      <td>Oct 05 2020</td>\n",
       "      <td>Dec 22 2020</td>\n",
       "      <td>RNA</td>\n",
       "      <td>1</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>9606</td>\n",
       "      <td>...</td>\n",
       "      <td>Maider,,MuÃ±oz-Culla</td>\n",
       "      <td>maider.munoz@ehu.eus</td>\n",
       "      <td>Biodonostia Institute</td>\n",
       "      <td>Paseo Dr.Begiristain s/n</td>\n",
       "      <td>San Sebastian</td>\n",
       "      <td>20014</td>\n",
       "      <td>Spain</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4817...</td>\n",
       "      <td>GSE159033,GSE159036</td>\n",
       "      <td>6631</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM4817994</th>\n",
       "      <td>Sample02-negative</td>\n",
       "      <td>GSM4817994</td>\n",
       "      <td>Public on Dec 21 2020</td>\n",
       "      <td>Oct 05 2020</td>\n",
       "      <td>Dec 22 2020</td>\n",
       "      <td>RNA</td>\n",
       "      <td>1</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>9606</td>\n",
       "      <td>...</td>\n",
       "      <td>Maider,,MuÃ±oz-Culla</td>\n",
       "      <td>maider.munoz@ehu.eus</td>\n",
       "      <td>Biodonostia Institute</td>\n",
       "      <td>Paseo Dr.Begiristain s/n</td>\n",
       "      <td>San Sebastian</td>\n",
       "      <td>20014</td>\n",
       "      <td>Spain</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4817...</td>\n",
       "      <td>GSE159033,GSE159036</td>\n",
       "      <td>6631</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM4817995</th>\n",
       "      <td>Sample06-negative</td>\n",
       "      <td>GSM4817995</td>\n",
       "      <td>Public on Dec 21 2020</td>\n",
       "      <td>Oct 05 2020</td>\n",
       "      <td>Dec 22 2020</td>\n",
       "      <td>RNA</td>\n",
       "      <td>1</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>9606</td>\n",
       "      <td>...</td>\n",
       "      <td>Maider,,MuÃ±oz-Culla</td>\n",
       "      <td>maider.munoz@ehu.eus</td>\n",
       "      <td>Biodonostia Institute</td>\n",
       "      <td>Paseo Dr.Begiristain s/n</td>\n",
       "      <td>San Sebastian</td>\n",
       "      <td>20014</td>\n",
       "      <td>Spain</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4817...</td>\n",
       "      <td>GSE159033,GSE159036</td>\n",
       "      <td>6631</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM4817996</th>\n",
       "      <td>Sample07-negative</td>\n",
       "      <td>GSM4817996</td>\n",
       "      <td>Public on Dec 21 2020</td>\n",
       "      <td>Oct 05 2020</td>\n",
       "      <td>Dec 22 2020</td>\n",
       "      <td>RNA</td>\n",
       "      <td>1</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>9606</td>\n",
       "      <td>...</td>\n",
       "      <td>Maider,,MuÃ±oz-Culla</td>\n",
       "      <td>maider.munoz@ehu.eus</td>\n",
       "      <td>Biodonostia Institute</td>\n",
       "      <td>Paseo Dr.Begiristain s/n</td>\n",
       "      <td>San Sebastian</td>\n",
       "      <td>20014</td>\n",
       "      <td>Spain</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4817...</td>\n",
       "      <td>GSE159033,GSE159036</td>\n",
       "      <td>6631</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM4817997</th>\n",
       "      <td>Sample09-negative</td>\n",
       "      <td>GSM4817997</td>\n",
       "      <td>Public on Dec 21 2020</td>\n",
       "      <td>Oct 05 2020</td>\n",
       "      <td>Dec 22 2020</td>\n",
       "      <td>RNA</td>\n",
       "      <td>1</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>9606</td>\n",
       "      <td>...</td>\n",
       "      <td>Maider,,MuÃ±oz-Culla</td>\n",
       "      <td>maider.munoz@ehu.eus</td>\n",
       "      <td>Biodonostia Institute</td>\n",
       "      <td>Paseo Dr.Begiristain s/n</td>\n",
       "      <td>San Sebastian</td>\n",
       "      <td>20014</td>\n",
       "      <td>Spain</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4817...</td>\n",
       "      <td>GSE159033,GSE159036</td>\n",
       "      <td>6631</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM4817998</th>\n",
       "      <td>Sample10-negative</td>\n",
       "      <td>GSM4817998</td>\n",
       "      <td>Public on Dec 21 2020</td>\n",
       "      <td>Oct 05 2020</td>\n",
       "      <td>Dec 22 2020</td>\n",
       "      <td>RNA</td>\n",
       "      <td>1</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>9606</td>\n",
       "      <td>...</td>\n",
       "      <td>Maider,,MuÃ±oz-Culla</td>\n",
       "      <td>maider.munoz@ehu.eus</td>\n",
       "      <td>Biodonostia Institute</td>\n",
       "      <td>Paseo Dr.Begiristain s/n</td>\n",
       "      <td>San Sebastian</td>\n",
       "      <td>20014</td>\n",
       "      <td>Spain</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4817...</td>\n",
       "      <td>GSE159033,GSE159036</td>\n",
       "      <td>6631</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM4817999</th>\n",
       "      <td>Sample15-negative</td>\n",
       "      <td>GSM4817999</td>\n",
       "      <td>Public on Dec 21 2020</td>\n",
       "      <td>Oct 05 2020</td>\n",
       "      <td>Dec 22 2020</td>\n",
       "      <td>RNA</td>\n",
       "      <td>1</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>9606</td>\n",
       "      <td>...</td>\n",
       "      <td>Maider,,MuÃ±oz-Culla</td>\n",
       "      <td>maider.munoz@ehu.eus</td>\n",
       "      <td>Biodonostia Institute</td>\n",
       "      <td>Paseo Dr.Begiristain s/n</td>\n",
       "      <td>San Sebastian</td>\n",
       "      <td>20014</td>\n",
       "      <td>Spain</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4817...</td>\n",
       "      <td>GSE159033,GSE159036</td>\n",
       "      <td>6631</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11 rows × 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        title geo_accession                 status  \\\n",
       "GSM4817989  Sample04-positive    GSM4817989  Public on Dec 21 2020   \n",
       "GSM4817990  Sample05-positive    GSM4817990  Public on Dec 21 2020   \n",
       "GSM4817991  Sample11-positive    GSM4817991  Public on Dec 21 2020   \n",
       "GSM4817992  Sample14-positive    GSM4817992  Public on Dec 21 2020   \n",
       "GSM4817993  Sample19-positive    GSM4817993  Public on Dec 21 2020   \n",
       "GSM4817994  Sample02-negative    GSM4817994  Public on Dec 21 2020   \n",
       "GSM4817995  Sample06-negative    GSM4817995  Public on Dec 21 2020   \n",
       "GSM4817996  Sample07-negative    GSM4817996  Public on Dec 21 2020   \n",
       "GSM4817997  Sample09-negative    GSM4817997  Public on Dec 21 2020   \n",
       "GSM4817998  Sample10-negative    GSM4817998  Public on Dec 21 2020   \n",
       "GSM4817999  Sample15-negative    GSM4817999  Public on Dec 21 2020   \n",
       "\n",
       "           submission_date last_update_date type channel_count  \\\n",
       "GSM4817989     Oct 05 2020      Dec 22 2020  RNA             1   \n",
       "GSM4817990     Oct 05 2020      Dec 22 2020  RNA             1   \n",
       "GSM4817991     Oct 05 2020      Dec 22 2020  RNA             1   \n",
       "GSM4817992     Oct 05 2020      Dec 22 2020  RNA             1   \n",
       "GSM4817993     Oct 05 2020      Dec 22 2020  RNA             1   \n",
       "GSM4817994     Oct 05 2020      Dec 22 2020  RNA             1   \n",
       "GSM4817995     Oct 05 2020      Dec 22 2020  RNA             1   \n",
       "GSM4817996     Oct 05 2020      Dec 22 2020  RNA             1   \n",
       "GSM4817997     Oct 05 2020      Dec 22 2020  RNA             1   \n",
       "GSM4817998     Oct 05 2020      Dec 22 2020  RNA             1   \n",
       "GSM4817999     Oct 05 2020      Dec 22 2020  RNA             1   \n",
       "\n",
       "           source_name_ch1  organism_ch1 taxid_ch1  ...          contact_name  \\\n",
       "GSM4817989           Blood  Homo sapiens      9606  ...  Maider,,MuÃ±oz-Culla   \n",
       "GSM4817990           Blood  Homo sapiens      9606  ...  Maider,,MuÃ±oz-Culla   \n",
       "GSM4817991           Blood  Homo sapiens      9606  ...  Maider,,MuÃ±oz-Culla   \n",
       "GSM4817992           Blood  Homo sapiens      9606  ...  Maider,,MuÃ±oz-Culla   \n",
       "GSM4817993           Blood  Homo sapiens      9606  ...  Maider,,MuÃ±oz-Culla   \n",
       "GSM4817994           Blood  Homo sapiens      9606  ...  Maider,,MuÃ±oz-Culla   \n",
       "GSM4817995           Blood  Homo sapiens      9606  ...  Maider,,MuÃ±oz-Culla   \n",
       "GSM4817996           Blood  Homo sapiens      9606  ...  Maider,,MuÃ±oz-Culla   \n",
       "GSM4817997           Blood  Homo sapiens      9606  ...  Maider,,MuÃ±oz-Culla   \n",
       "GSM4817998           Blood  Homo sapiens      9606  ...  Maider,,MuÃ±oz-Culla   \n",
       "GSM4817999           Blood  Homo sapiens      9606  ...  Maider,,MuÃ±oz-Culla   \n",
       "\n",
       "                   contact_email      contact_institute  \\\n",
       "GSM4817989  maider.munoz@ehu.eus  Biodonostia Institute   \n",
       "GSM4817990  maider.munoz@ehu.eus  Biodonostia Institute   \n",
       "GSM4817991  maider.munoz@ehu.eus  Biodonostia Institute   \n",
       "GSM4817992  maider.munoz@ehu.eus  Biodonostia Institute   \n",
       "GSM4817993  maider.munoz@ehu.eus  Biodonostia Institute   \n",
       "GSM4817994  maider.munoz@ehu.eus  Biodonostia Institute   \n",
       "GSM4817995  maider.munoz@ehu.eus  Biodonostia Institute   \n",
       "GSM4817996  maider.munoz@ehu.eus  Biodonostia Institute   \n",
       "GSM4817997  maider.munoz@ehu.eus  Biodonostia Institute   \n",
       "GSM4817998  maider.munoz@ehu.eus  Biodonostia Institute   \n",
       "GSM4817999  maider.munoz@ehu.eus  Biodonostia Institute   \n",
       "\n",
       "                     contact_address   contact_city contact_zip/postal_code  \\\n",
       "GSM4817989  Paseo Dr.Begiristain s/n  San Sebastian                   20014   \n",
       "GSM4817990  Paseo Dr.Begiristain s/n  San Sebastian                   20014   \n",
       "GSM4817991  Paseo Dr.Begiristain s/n  San Sebastian                   20014   \n",
       "GSM4817992  Paseo Dr.Begiristain s/n  San Sebastian                   20014   \n",
       "GSM4817993  Paseo Dr.Begiristain s/n  San Sebastian                   20014   \n",
       "GSM4817994  Paseo Dr.Begiristain s/n  San Sebastian                   20014   \n",
       "GSM4817995  Paseo Dr.Begiristain s/n  San Sebastian                   20014   \n",
       "GSM4817996  Paseo Dr.Begiristain s/n  San Sebastian                   20014   \n",
       "GSM4817997  Paseo Dr.Begiristain s/n  San Sebastian                   20014   \n",
       "GSM4817998  Paseo Dr.Begiristain s/n  San Sebastian                   20014   \n",
       "GSM4817999  Paseo Dr.Begiristain s/n  San Sebastian                   20014   \n",
       "\n",
       "           contact_country                                 supplementary_file  \\\n",
       "GSM4817989           Spain  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4817...   \n",
       "GSM4817990           Spain  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4817...   \n",
       "GSM4817991           Spain  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4817...   \n",
       "GSM4817992           Spain  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4817...   \n",
       "GSM4817993           Spain  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4817...   \n",
       "GSM4817994           Spain  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4817...   \n",
       "GSM4817995           Spain  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4817...   \n",
       "GSM4817996           Spain  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4817...   \n",
       "GSM4817997           Spain  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4817...   \n",
       "GSM4817998           Spain  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4817...   \n",
       "GSM4817999           Spain  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4817...   \n",
       "\n",
       "                      series_id data_row_count  \n",
       "GSM4817989  GSE159033,GSE159036           6631  \n",
       "GSM4817990  GSE159033,GSE159036           6631  \n",
       "GSM4817991  GSE159033,GSE159036           6631  \n",
       "GSM4817992  GSE159033,GSE159036           6631  \n",
       "GSM4817993  GSE159033,GSE159036           6631  \n",
       "GSM4817994  GSE159033,GSE159036           6631  \n",
       "GSM4817995  GSE159033,GSE159036           6631  \n",
       "GSM4817996  GSE159033,GSE159036           6631  \n",
       "GSM4817997  GSE159033,GSE159036           6631  \n",
       "GSM4817998  GSE159033,GSE159036           6631  \n",
       "GSM4817999  GSE159033,GSE159036           6631  \n",
       "\n",
       "[11 rows x 32 columns]"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gse_family_df = gse_family.phenotype_data\n",
    "gse_family_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [],
   "source": [
    "gse_family_df.to_csv('GSE159033_family_phenotype_data.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [],
   "source": [
    "SERIES_MATRIX_PATH = 'GSE159033/GSE159033_series_matrix.txt/GSE159033_series_matrix.txt'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID_REF</th>\n",
       "      <th>GSM4817989</th>\n",
       "      <th>GSM4817990</th>\n",
       "      <th>GSM4817991</th>\n",
       "      <th>GSM4817992</th>\n",
       "      <th>GSM4817993</th>\n",
       "      <th>GSM4817994</th>\n",
       "      <th>GSM4817995</th>\n",
       "      <th>GSM4817996</th>\n",
       "      <th>GSM4817997</th>\n",
       "      <th>GSM4817998</th>\n",
       "      <th>GSM4817999</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20500112</td>\n",
       "      <td>12.08</td>\n",
       "      <td>10.98</td>\n",
       "      <td>12.64</td>\n",
       "      <td>11.17</td>\n",
       "      <td>11.89</td>\n",
       "      <td>11.50</td>\n",
       "      <td>11.56</td>\n",
       "      <td>11.73</td>\n",
       "      <td>11.55</td>\n",
       "      <td>12.16</td>\n",
       "      <td>11.64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20500113</td>\n",
       "      <td>1.60</td>\n",
       "      <td>1.34</td>\n",
       "      <td>0.89</td>\n",
       "      <td>1.11</td>\n",
       "      <td>1.14</td>\n",
       "      <td>0.77</td>\n",
       "      <td>0.96</td>\n",
       "      <td>0.79</td>\n",
       "      <td>1.21</td>\n",
       "      <td>1.35</td>\n",
       "      <td>1.38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20500114</td>\n",
       "      <td>1.10</td>\n",
       "      <td>1.32</td>\n",
       "      <td>0.86</td>\n",
       "      <td>0.92</td>\n",
       "      <td>0.99</td>\n",
       "      <td>1.18</td>\n",
       "      <td>0.77</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.51</td>\n",
       "      <td>0.73</td>\n",
       "      <td>0.51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20500115</td>\n",
       "      <td>12.45</td>\n",
       "      <td>13.46</td>\n",
       "      <td>12.67</td>\n",
       "      <td>12.41</td>\n",
       "      <td>12.94</td>\n",
       "      <td>12.43</td>\n",
       "      <td>12.64</td>\n",
       "      <td>12.60</td>\n",
       "      <td>12.00</td>\n",
       "      <td>12.64</td>\n",
       "      <td>12.80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20500116</td>\n",
       "      <td>1.69</td>\n",
       "      <td>1.66</td>\n",
       "      <td>1.44</td>\n",
       "      <td>0.86</td>\n",
       "      <td>1.33</td>\n",
       "      <td>1.44</td>\n",
       "      <td>1.46</td>\n",
       "      <td>0.85</td>\n",
       "      <td>0.89</td>\n",
       "      <td>1.44</td>\n",
       "      <td>2.22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6626</th>\n",
       "      <td>20538487</td>\n",
       "      <td>13.05</td>\n",
       "      <td>13.53</td>\n",
       "      <td>13.11</td>\n",
       "      <td>13.38</td>\n",
       "      <td>13.25</td>\n",
       "      <td>13.31</td>\n",
       "      <td>13.66</td>\n",
       "      <td>13.35</td>\n",
       "      <td>12.82</td>\n",
       "      <td>13.01</td>\n",
       "      <td>13.44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6627</th>\n",
       "      <td>20538488</td>\n",
       "      <td>13.10</td>\n",
       "      <td>13.73</td>\n",
       "      <td>13.16</td>\n",
       "      <td>13.39</td>\n",
       "      <td>13.31</td>\n",
       "      <td>13.33</td>\n",
       "      <td>13.66</td>\n",
       "      <td>13.36</td>\n",
       "      <td>12.76</td>\n",
       "      <td>13.08</td>\n",
       "      <td>13.57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6628</th>\n",
       "      <td>20538489</td>\n",
       "      <td>13.02</td>\n",
       "      <td>13.73</td>\n",
       "      <td>13.05</td>\n",
       "      <td>13.23</td>\n",
       "      <td>13.27</td>\n",
       "      <td>13.30</td>\n",
       "      <td>13.49</td>\n",
       "      <td>13.43</td>\n",
       "      <td>12.79</td>\n",
       "      <td>12.88</td>\n",
       "      <td>13.44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6629</th>\n",
       "      <td>20538490</td>\n",
       "      <td>12.99</td>\n",
       "      <td>13.56</td>\n",
       "      <td>13.13</td>\n",
       "      <td>13.34</td>\n",
       "      <td>13.28</td>\n",
       "      <td>13.27</td>\n",
       "      <td>13.50</td>\n",
       "      <td>13.22</td>\n",
       "      <td>12.79</td>\n",
       "      <td>12.88</td>\n",
       "      <td>13.40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6630</th>\n",
       "      <td>20538491</td>\n",
       "      <td>13.10</td>\n",
       "      <td>13.73</td>\n",
       "      <td>13.05</td>\n",
       "      <td>13.36</td>\n",
       "      <td>13.30</td>\n",
       "      <td>13.27</td>\n",
       "      <td>13.54</td>\n",
       "      <td>13.36</td>\n",
       "      <td>12.78</td>\n",
       "      <td>12.98</td>\n",
       "      <td>13.47</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6631 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        ID_REF  GSM4817989  GSM4817990  GSM4817991  GSM4817992  GSM4817993  \\\n",
       "0     20500112       12.08       10.98       12.64       11.17       11.89   \n",
       "1     20500113        1.60        1.34        0.89        1.11        1.14   \n",
       "2     20500114        1.10        1.32        0.86        0.92        0.99   \n",
       "3     20500115       12.45       13.46       12.67       12.41       12.94   \n",
       "4     20500116        1.69        1.66        1.44        0.86        1.33   \n",
       "...        ...         ...         ...         ...         ...         ...   \n",
       "6626  20538487       13.05       13.53       13.11       13.38       13.25   \n",
       "6627  20538488       13.10       13.73       13.16       13.39       13.31   \n",
       "6628  20538489       13.02       13.73       13.05       13.23       13.27   \n",
       "6629  20538490       12.99       13.56       13.13       13.34       13.28   \n",
       "6630  20538491       13.10       13.73       13.05       13.36       13.30   \n",
       "\n",
       "      GSM4817994  GSM4817995  GSM4817996  GSM4817997  GSM4817998  GSM4817999  \n",
       "0          11.50       11.56       11.73       11.55       12.16       11.64  \n",
       "1           0.77        0.96        0.79        1.21        1.35        1.38  \n",
       "2           1.18        0.77        0.97        0.51        0.73        0.51  \n",
       "3          12.43       12.64       12.60       12.00       12.64       12.80  \n",
       "4           1.44        1.46        0.85        0.89        1.44        2.22  \n",
       "...          ...         ...         ...         ...         ...         ...  \n",
       "6626       13.31       13.66       13.35       12.82       13.01       13.44  \n",
       "6627       13.33       13.66       13.36       12.76       13.08       13.57  \n",
       "6628       13.30       13.49       13.43       12.79       12.88       13.44  \n",
       "6629       13.27       13.50       13.22       12.79       12.88       13.40  \n",
       "6630       13.27       13.54       13.36       12.78       12.98       13.47  \n",
       "\n",
       "[6631 rows x 12 columns]"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "series_df = pd.read_csv(SERIES_MATRIX_PATH, sep='\\t', header=66, low_memory=False)\n",
    "series_df = series_df.drop(index=6631)\n",
    "series_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Concatenate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\giuli\\AppData\\Local\\Temp\\ipykernel_5516\\766904052.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  miRNA_homo_sapiens_df['ID'] = miRNA_homo_sapiens_df['ID'].astype(str)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Accession_ID</th>\n",
       "      <th>Accession</th>\n",
       "      <th>Transcript ID(Array Design)</th>\n",
       "      <th>Sequence Type</th>\n",
       "      <th>Species Scientific Name</th>\n",
       "      <th>Alignments</th>\n",
       "      <th>Sequence Length</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Genome Context</th>\n",
       "      <th>...</th>\n",
       "      <th>GSM4817990</th>\n",
       "      <th>GSM4817991</th>\n",
       "      <th>GSM4817992</th>\n",
       "      <th>GSM4817993</th>\n",
       "      <th>GSM4817994</th>\n",
       "      <th>GSM4817995</th>\n",
       "      <th>GSM4817996</th>\n",
       "      <th>GSM4817997</th>\n",
       "      <th>GSM4817998</th>\n",
       "      <th>GSM4817999</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20500112</td>\n",
       "      <td>MIMAT0000062_st</td>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>miRNA</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>chr11:122017276-122017297 (-) /// chr22:465086...</td>\n",
       "      <td>22</td>\n",
       "      <td>UGAGGUAGUAGGUUGUAUAGUU</td>\n",
       "      <td>ENST00000362295 // sense // exon // 1 /// ENST...</td>\n",
       "      <td>...</td>\n",
       "      <td>10.98</td>\n",
       "      <td>12.64</td>\n",
       "      <td>11.17</td>\n",
       "      <td>11.89</td>\n",
       "      <td>11.50</td>\n",
       "      <td>11.56</td>\n",
       "      <td>11.73</td>\n",
       "      <td>11.55</td>\n",
       "      <td>12.16</td>\n",
       "      <td>11.64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20500113</td>\n",
       "      <td>MIMAT0004481_st</td>\n",
       "      <td>MIMAT0004481</td>\n",
       "      <td>hsa-let-7a-3p</td>\n",
       "      <td>miRNA</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>chr22:46508680-46508700 (+) /// chr9:96938295-...</td>\n",
       "      <td>21</td>\n",
       "      <td>CUAUACAAUCUACUGUCUUUC</td>\n",
       "      <td>ENST00000362295 // sense // exon // 1 /// ENST...</td>\n",
       "      <td>...</td>\n",
       "      <td>1.34</td>\n",
       "      <td>0.89</td>\n",
       "      <td>1.11</td>\n",
       "      <td>1.14</td>\n",
       "      <td>0.77</td>\n",
       "      <td>0.96</td>\n",
       "      <td>0.79</td>\n",
       "      <td>1.21</td>\n",
       "      <td>1.35</td>\n",
       "      <td>1.38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20500114</td>\n",
       "      <td>MIMAT0010195_st</td>\n",
       "      <td>MIMAT0010195</td>\n",
       "      <td>hsa-let-7a-2-3p</td>\n",
       "      <td>miRNA</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>chr11:122017231-122017252 (-)</td>\n",
       "      <td>22</td>\n",
       "      <td>CUGUACAGCCUCCUAGCUUUCC</td>\n",
       "      <td>ENST00000362105 // sense // exon // 1</td>\n",
       "      <td>...</td>\n",
       "      <td>1.32</td>\n",
       "      <td>0.86</td>\n",
       "      <td>0.92</td>\n",
       "      <td>0.99</td>\n",
       "      <td>1.18</td>\n",
       "      <td>0.77</td>\n",
       "      <td>0.97</td>\n",
       "      <td>0.51</td>\n",
       "      <td>0.73</td>\n",
       "      <td>0.51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20500115</td>\n",
       "      <td>MIMAT0000063_st</td>\n",
       "      <td>MIMAT0000063</td>\n",
       "      <td>hsa-let-7b-5p</td>\n",
       "      <td>miRNA</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>chr22:46509571-46509592 (+)</td>\n",
       "      <td>22</td>\n",
       "      <td>UGAGGUAGUAGGUUGUGUGGUU</td>\n",
       "      <td>ENST00000360737 // sense // exon // 5 /// OTTH...</td>\n",
       "      <td>...</td>\n",
       "      <td>13.46</td>\n",
       "      <td>12.67</td>\n",
       "      <td>12.41</td>\n",
       "      <td>12.94</td>\n",
       "      <td>12.43</td>\n",
       "      <td>12.64</td>\n",
       "      <td>12.60</td>\n",
       "      <td>12.00</td>\n",
       "      <td>12.64</td>\n",
       "      <td>12.80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20500116</td>\n",
       "      <td>MIMAT0004482_st</td>\n",
       "      <td>MIMAT0004482</td>\n",
       "      <td>hsa-let-7b-3p</td>\n",
       "      <td>miRNA</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>chr22:46509625-46509646 (+)</td>\n",
       "      <td>22</td>\n",
       "      <td>CUAUACAACCUACUGCCUUCCC</td>\n",
       "      <td>ENST00000360737 // sense // exon // 5 /// OTTH...</td>\n",
       "      <td>...</td>\n",
       "      <td>1.66</td>\n",
       "      <td>1.44</td>\n",
       "      <td>0.86</td>\n",
       "      <td>1.33</td>\n",
       "      <td>1.44</td>\n",
       "      <td>1.46</td>\n",
       "      <td>0.85</td>\n",
       "      <td>0.89</td>\n",
       "      <td>1.44</td>\n",
       "      <td>2.22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6626</th>\n",
       "      <td>20538487</td>\n",
       "      <td>gi555853_copy5_st</td>\n",
       "      <td>gi555853_copy5</td>\n",
       "      <td>gi:555853</td>\n",
       "      <td>5.8s rRNA</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>---</td>\n",
       "      <td>157</td>\n",
       "      <td>CGACTCTTAGCGGTGGATCACTCGGCTCGTGCGTCGATGAAGAACG...</td>\n",
       "      <td>---</td>\n",
       "      <td>...</td>\n",
       "      <td>13.53</td>\n",
       "      <td>13.11</td>\n",
       "      <td>13.38</td>\n",
       "      <td>13.25</td>\n",
       "      <td>13.31</td>\n",
       "      <td>13.66</td>\n",
       "      <td>13.35</td>\n",
       "      <td>12.82</td>\n",
       "      <td>13.01</td>\n",
       "      <td>13.44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6627</th>\n",
       "      <td>20538488</td>\n",
       "      <td>gi555853_copy6_st</td>\n",
       "      <td>gi555853_copy6</td>\n",
       "      <td>gi:555853</td>\n",
       "      <td>5.8s rRNA</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>---</td>\n",
       "      <td>157</td>\n",
       "      <td>CGACTCTTAGCGGTGGATCACTCGGCTCGTGCGTCGATGAAGAACG...</td>\n",
       "      <td>---</td>\n",
       "      <td>...</td>\n",
       "      <td>13.73</td>\n",
       "      <td>13.16</td>\n",
       "      <td>13.39</td>\n",
       "      <td>13.31</td>\n",
       "      <td>13.33</td>\n",
       "      <td>13.66</td>\n",
       "      <td>13.36</td>\n",
       "      <td>12.76</td>\n",
       "      <td>13.08</td>\n",
       "      <td>13.57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6628</th>\n",
       "      <td>20538489</td>\n",
       "      <td>gi555853_copy7_st</td>\n",
       "      <td>gi555853_copy7</td>\n",
       "      <td>gi:555853</td>\n",
       "      <td>5.8s rRNA</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>---</td>\n",
       "      <td>157</td>\n",
       "      <td>CGACTCTTAGCGGTGGATCACTCGGCTCGTGCGTCGATGAAGAACG...</td>\n",
       "      <td>---</td>\n",
       "      <td>...</td>\n",
       "      <td>13.73</td>\n",
       "      <td>13.05</td>\n",
       "      <td>13.23</td>\n",
       "      <td>13.27</td>\n",
       "      <td>13.30</td>\n",
       "      <td>13.49</td>\n",
       "      <td>13.43</td>\n",
       "      <td>12.79</td>\n",
       "      <td>12.88</td>\n",
       "      <td>13.44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6629</th>\n",
       "      <td>20538490</td>\n",
       "      <td>gi555853_copy8_st</td>\n",
       "      <td>gi555853_copy8</td>\n",
       "      <td>gi:555853</td>\n",
       "      <td>5.8s rRNA</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>---</td>\n",
       "      <td>157</td>\n",
       "      <td>CGACTCTTAGCGGTGGATCACTCGGCTCGTGCGTCGATGAAGAACG...</td>\n",
       "      <td>---</td>\n",
       "      <td>...</td>\n",
       "      <td>13.56</td>\n",
       "      <td>13.13</td>\n",
       "      <td>13.34</td>\n",
       "      <td>13.28</td>\n",
       "      <td>13.27</td>\n",
       "      <td>13.50</td>\n",
       "      <td>13.22</td>\n",
       "      <td>12.79</td>\n",
       "      <td>12.88</td>\n",
       "      <td>13.40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6630</th>\n",
       "      <td>20538491</td>\n",
       "      <td>gi555853_copy9_st</td>\n",
       "      <td>gi555853_copy9</td>\n",
       "      <td>gi:555853</td>\n",
       "      <td>5.8s rRNA</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>---</td>\n",
       "      <td>157</td>\n",
       "      <td>CGACTCTTAGCGGTGGATCACTCGGCTCGTGCGTCGATGAAGAACG...</td>\n",
       "      <td>---</td>\n",
       "      <td>...</td>\n",
       "      <td>13.73</td>\n",
       "      <td>13.05</td>\n",
       "      <td>13.36</td>\n",
       "      <td>13.30</td>\n",
       "      <td>13.27</td>\n",
       "      <td>13.54</td>\n",
       "      <td>13.36</td>\n",
       "      <td>12.78</td>\n",
       "      <td>12.98</td>\n",
       "      <td>13.47</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6631 rows × 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            ID       Accession_ID       Accession Transcript ID(Array Design)  \\\n",
       "0     20500112    MIMAT0000062_st    MIMAT0000062               hsa-let-7a-5p   \n",
       "1     20500113    MIMAT0004481_st    MIMAT0004481               hsa-let-7a-3p   \n",
       "2     20500114    MIMAT0010195_st    MIMAT0010195             hsa-let-7a-2-3p   \n",
       "3     20500115    MIMAT0000063_st    MIMAT0000063               hsa-let-7b-5p   \n",
       "4     20500116    MIMAT0004482_st    MIMAT0004482               hsa-let-7b-3p   \n",
       "...        ...                ...             ...                         ...   \n",
       "6626  20538487  gi555853_copy5_st  gi555853_copy5                   gi:555853   \n",
       "6627  20538488  gi555853_copy6_st  gi555853_copy6                   gi:555853   \n",
       "6628  20538489  gi555853_copy7_st  gi555853_copy7                   gi:555853   \n",
       "6629  20538490  gi555853_copy8_st  gi555853_copy8                   gi:555853   \n",
       "6630  20538491  gi555853_copy9_st  gi555853_copy9                   gi:555853   \n",
       "\n",
       "     Sequence Type Species Scientific Name  \\\n",
       "0            miRNA            Homo sapiens   \n",
       "1            miRNA            Homo sapiens   \n",
       "2            miRNA            Homo sapiens   \n",
       "3            miRNA            Homo sapiens   \n",
       "4            miRNA            Homo sapiens   \n",
       "...            ...                     ...   \n",
       "6626     5.8s rRNA            Homo sapiens   \n",
       "6627     5.8s rRNA            Homo sapiens   \n",
       "6628     5.8s rRNA            Homo sapiens   \n",
       "6629     5.8s rRNA            Homo sapiens   \n",
       "6630     5.8s rRNA            Homo sapiens   \n",
       "\n",
       "                                             Alignments Sequence Length  \\\n",
       "0     chr11:122017276-122017297 (-) /// chr22:465086...              22   \n",
       "1     chr22:46508680-46508700 (+) /// chr9:96938295-...              21   \n",
       "2                         chr11:122017231-122017252 (-)              22   \n",
       "3                           chr22:46509571-46509592 (+)              22   \n",
       "4                           chr22:46509625-46509646 (+)              22   \n",
       "...                                                 ...             ...   \n",
       "6626                                                ---             157   \n",
       "6627                                                ---             157   \n",
       "6628                                                ---             157   \n",
       "6629                                                ---             157   \n",
       "6630                                                ---             157   \n",
       "\n",
       "                                               Sequence  \\\n",
       "0                                UGAGGUAGUAGGUUGUAUAGUU   \n",
       "1                                 CUAUACAAUCUACUGUCUUUC   \n",
       "2                                CUGUACAGCCUCCUAGCUUUCC   \n",
       "3                                UGAGGUAGUAGGUUGUGUGGUU   \n",
       "4                                CUAUACAACCUACUGCCUUCCC   \n",
       "...                                                 ...   \n",
       "6626  CGACTCTTAGCGGTGGATCACTCGGCTCGTGCGTCGATGAAGAACG...   \n",
       "6627  CGACTCTTAGCGGTGGATCACTCGGCTCGTGCGTCGATGAAGAACG...   \n",
       "6628  CGACTCTTAGCGGTGGATCACTCGGCTCGTGCGTCGATGAAGAACG...   \n",
       "6629  CGACTCTTAGCGGTGGATCACTCGGCTCGTGCGTCGATGAAGAACG...   \n",
       "6630  CGACTCTTAGCGGTGGATCACTCGGCTCGTGCGTCGATGAAGAACG...   \n",
       "\n",
       "                                         Genome Context  ... GSM4817990  \\\n",
       "0     ENST00000362295 // sense // exon // 1 /// ENST...  ...      10.98   \n",
       "1     ENST00000362295 // sense // exon // 1 /// ENST...  ...       1.34   \n",
       "2                 ENST00000362105 // sense // exon // 1  ...       1.32   \n",
       "3     ENST00000360737 // sense // exon // 5 /// OTTH...  ...      13.46   \n",
       "4     ENST00000360737 // sense // exon // 5 /// OTTH...  ...       1.66   \n",
       "...                                                 ...  ...        ...   \n",
       "6626                                                ---  ...      13.53   \n",
       "6627                                                ---  ...      13.73   \n",
       "6628                                                ---  ...      13.73   \n",
       "6629                                                ---  ...      13.56   \n",
       "6630                                                ---  ...      13.73   \n",
       "\n",
       "     GSM4817991 GSM4817992 GSM4817993 GSM4817994 GSM4817995 GSM4817996  \\\n",
       "0         12.64      11.17      11.89      11.50      11.56      11.73   \n",
       "1          0.89       1.11       1.14       0.77       0.96       0.79   \n",
       "2          0.86       0.92       0.99       1.18       0.77       0.97   \n",
       "3         12.67      12.41      12.94      12.43      12.64      12.60   \n",
       "4          1.44       0.86       1.33       1.44       1.46       0.85   \n",
       "...         ...        ...        ...        ...        ...        ...   \n",
       "6626      13.11      13.38      13.25      13.31      13.66      13.35   \n",
       "6627      13.16      13.39      13.31      13.33      13.66      13.36   \n",
       "6628      13.05      13.23      13.27      13.30      13.49      13.43   \n",
       "6629      13.13      13.34      13.28      13.27      13.50      13.22   \n",
       "6630      13.05      13.36      13.30      13.27      13.54      13.36   \n",
       "\n",
       "     GSM4817997  GSM4817998  GSM4817999  \n",
       "0         11.55       12.16       11.64  \n",
       "1          1.21        1.35        1.38  \n",
       "2          0.51        0.73        0.51  \n",
       "3         12.00       12.64       12.80  \n",
       "4          0.89        1.44        2.22  \n",
       "...         ...         ...         ...  \n",
       "6626      12.82       13.01       13.44  \n",
       "6627      12.76       13.08       13.57  \n",
       "6628      12.79       12.88       13.44  \n",
       "6629      12.79       12.88       13.40  \n",
       "6630      12.78       12.98       13.47  \n",
       "\n",
       "[6631 rows x 29 columns]"
      ]
     },
     "execution_count": 108,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "miRNA_homo_sapiens_df['ID'] = miRNA_homo_sapiens_df['ID'].astype(str)\n",
    "series_df['ID_REF'] = series_df['ID_REF'].astype(str)\n",
    "\n",
    "merged_df = miRNA_homo_sapiens_df.merge(series_df, left_on='ID', right_on='ID_REF', how='inner')\n",
    "merged_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['ID_REF', 'GSM4817989', 'GSM4817990', 'GSM4817991', 'GSM4817992',\n",
       "       'GSM4817993', 'GSM4817994', 'GSM4817995', 'GSM4817996', 'GSM4817997',\n",
       "       'GSM4817998', 'GSM4817999', 'ID', 'Accession_ID', 'Accession',\n",
       "       'Transcript ID(Array Design)', 'Sequence Type',\n",
       "       'Species Scientific Name', 'Alignments', 'Sequence Length', 'Sequence',\n",
       "       'Genome Context', 'Clustered miRNAs within 10kb', 'Target Genes',\n",
       "       'GeneChip Array', 'Annotation Date', 'Sequence Source', 'miRNA_ID',\n",
       "       'SPOT_ID'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_df.to_csv('GSE159033_genomic_matrix_matrix.csv')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
