{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\Users\\giuli\\anaconda3\\lib\\site-packages\\pandas\\core\\arrays\\masked.py:60: UserWarning: Pandas requires version '1.3.6' or newer of 'bottleneck' (version '1.3.5' currently installed).\n",
      "  from pandas.core import (\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 1</th>\n",
       "      <th>Bisulfite conversion rate</th>\n",
       "      <th>Coverage</th>\n",
       "      <th>Depth</th>\n",
       "      <th>Bisulfite conversion rate.1</th>\n",
       "      <th>Coverage.1</th>\n",
       "      <th>Depth.1</th>\n",
       "      <th>Bisulfite conversion rate.2</th>\n",
       "      <th>Coverage.2</th>\n",
       "      <th>Depth.2</th>\n",
       "      <th>Bisulfite conversion rate.3</th>\n",
       "      <th>Coverage.3</th>\n",
       "      <th>Depth.3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HC</td>\n",
       "      <td>0.996512</td>\n",
       "      <td>0.930774</td>\n",
       "      <td>12.938460</td>\n",
       "      <td>0.996427</td>\n",
       "      <td>0.929601</td>\n",
       "      <td>17.736396</td>\n",
       "      <td>0.995716</td>\n",
       "      <td>0.929011</td>\n",
       "      <td>15.337275</td>\n",
       "      <td>0.996680</td>\n",
       "      <td>0.927372</td>\n",
       "      <td>13.229731</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HC</td>\n",
       "      <td>0.996372</td>\n",
       "      <td>0.930213</td>\n",
       "      <td>17.840417</td>\n",
       "      <td>0.996214</td>\n",
       "      <td>0.929735</td>\n",
       "      <td>17.822934</td>\n",
       "      <td>0.995485</td>\n",
       "      <td>0.929513</td>\n",
       "      <td>16.006528</td>\n",
       "      <td>0.996716</td>\n",
       "      <td>0.929211</td>\n",
       "      <td>16.705444</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HC</td>\n",
       "      <td>0.996646</td>\n",
       "      <td>0.930228</td>\n",
       "      <td>17.773436</td>\n",
       "      <td>0.996185</td>\n",
       "      <td>0.928996</td>\n",
       "      <td>16.137003</td>\n",
       "      <td>0.995200</td>\n",
       "      <td>0.930783</td>\n",
       "      <td>19.982329</td>\n",
       "      <td>0.996675</td>\n",
       "      <td>0.929492</td>\n",
       "      <td>16.934506</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HC</td>\n",
       "      <td>0.996599</td>\n",
       "      <td>0.930274</td>\n",
       "      <td>18.368304</td>\n",
       "      <td>0.996369</td>\n",
       "      <td>0.926824</td>\n",
       "      <td>12.071238</td>\n",
       "      <td>0.995258</td>\n",
       "      <td>0.929191</td>\n",
       "      <td>16.832175</td>\n",
       "      <td>0.996455</td>\n",
       "      <td>0.928490</td>\n",
       "      <td>13.884959</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>HC</td>\n",
       "      <td>0.996699</td>\n",
       "      <td>0.929747</td>\n",
       "      <td>17.377918</td>\n",
       "      <td>0.996125</td>\n",
       "      <td>0.929457</td>\n",
       "      <td>15.147257</td>\n",
       "      <td>0.995298</td>\n",
       "      <td>0.929502</td>\n",
       "      <td>16.636607</td>\n",
       "      <td>0.996767</td>\n",
       "      <td>0.929995</td>\n",
       "      <td>19.246635</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HC</td>\n",
       "      <td>0.996768</td>\n",
       "      <td>0.932192</td>\n",
       "      <td>23.408974</td>\n",
       "      <td>0.996068</td>\n",
       "      <td>0.928345</td>\n",
       "      <td>12.632632</td>\n",
       "      <td>0.995264</td>\n",
       "      <td>0.930080</td>\n",
       "      <td>14.346573</td>\n",
       "      <td>0.996661</td>\n",
       "      <td>0.928186</td>\n",
       "      <td>14.603209</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>HC</td>\n",
       "      <td>0.996947</td>\n",
       "      <td>0.929685</td>\n",
       "      <td>15.971407</td>\n",
       "      <td>0.995900</td>\n",
       "      <td>0.925510</td>\n",
       "      <td>9.288985</td>\n",
       "      <td>0.996674</td>\n",
       "      <td>0.929717</td>\n",
       "      <td>15.750571</td>\n",
       "      <td>0.996651</td>\n",
       "      <td>0.929517</td>\n",
       "      <td>14.622796</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>HC</td>\n",
       "      <td>0.996898</td>\n",
       "      <td>0.928607</td>\n",
       "      <td>13.548486</td>\n",
       "      <td>0.996175</td>\n",
       "      <td>0.926523</td>\n",
       "      <td>10.525559</td>\n",
       "      <td>0.996554</td>\n",
       "      <td>0.929193</td>\n",
       "      <td>14.511543</td>\n",
       "      <td>0.996471</td>\n",
       "      <td>0.927907</td>\n",
       "      <td>13.028801</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>HC</td>\n",
       "      <td>0.996820</td>\n",
       "      <td>0.930086</td>\n",
       "      <td>16.368358</td>\n",
       "      <td>0.996251</td>\n",
       "      <td>0.927231</td>\n",
       "      <td>10.924250</td>\n",
       "      <td>0.996321</td>\n",
       "      <td>0.930720</td>\n",
       "      <td>16.527435</td>\n",
       "      <td>0.996473</td>\n",
       "      <td>0.930439</td>\n",
       "      <td>17.994847</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>HC</td>\n",
       "      <td>0.996613</td>\n",
       "      <td>0.931072</td>\n",
       "      <td>19.442724</td>\n",
       "      <td>0.996034</td>\n",
       "      <td>0.931463</td>\n",
       "      <td>22.572802</td>\n",
       "      <td>0.996316</td>\n",
       "      <td>0.930101</td>\n",
       "      <td>15.576498</td>\n",
       "      <td>0.996176</td>\n",
       "      <td>0.926703</td>\n",
       "      <td>12.642628</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>HC</td>\n",
       "      <td>0.992790</td>\n",
       "      <td>0.931186</td>\n",
       "      <td>21.812578</td>\n",
       "      <td>0.992614</td>\n",
       "      <td>0.929490</td>\n",
       "      <td>17.864039</td>\n",
       "      <td>0.995146</td>\n",
       "      <td>0.930830</td>\n",
       "      <td>22.355377</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HC</td>\n",
       "      <td>0.992276</td>\n",
       "      <td>0.930920</td>\n",
       "      <td>20.684043</td>\n",
       "      <td>0.992878</td>\n",
       "      <td>0.930062</td>\n",
       "      <td>18.524907</td>\n",
       "      <td>0.994130</td>\n",
       "      <td>0.930743</td>\n",
       "      <td>22.457742</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>MS</td>\n",
       "      <td>0.996657</td>\n",
       "      <td>0.930324</td>\n",
       "      <td>17.984730</td>\n",
       "      <td>0.996089</td>\n",
       "      <td>0.929900</td>\n",
       "      <td>17.037665</td>\n",
       "      <td>0.995197</td>\n",
       "      <td>0.929483</td>\n",
       "      <td>17.773411</td>\n",
       "      <td>0.996676</td>\n",
       "      <td>0.928924</td>\n",
       "      <td>13.923285</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>MS</td>\n",
       "      <td>0.996421</td>\n",
       "      <td>0.930266</td>\n",
       "      <td>17.007209</td>\n",
       "      <td>0.996215</td>\n",
       "      <td>0.927086</td>\n",
       "      <td>12.921820</td>\n",
       "      <td>0.995429</td>\n",
       "      <td>0.928552</td>\n",
       "      <td>14.450575</td>\n",
       "      <td>0.996626</td>\n",
       "      <td>0.930401</td>\n",
       "      <td>19.157674</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>MS</td>\n",
       "      <td>0.996751</td>\n",
       "      <td>0.929358</td>\n",
       "      <td>15.815312</td>\n",
       "      <td>0.996331</td>\n",
       "      <td>0.929285</td>\n",
       "      <td>17.466062</td>\n",
       "      <td>0.995302</td>\n",
       "      <td>0.929864</td>\n",
       "      <td>17.509609</td>\n",
       "      <td>0.996735</td>\n",
       "      <td>0.928064</td>\n",
       "      <td>13.641882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>MS</td>\n",
       "      <td>0.996621</td>\n",
       "      <td>0.929229</td>\n",
       "      <td>14.935193</td>\n",
       "      <td>0.996349</td>\n",
       "      <td>0.928237</td>\n",
       "      <td>15.905936</td>\n",
       "      <td>0.995421</td>\n",
       "      <td>0.928881</td>\n",
       "      <td>15.297445</td>\n",
       "      <td>0.996731</td>\n",
       "      <td>0.928808</td>\n",
       "      <td>14.776338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>MS</td>\n",
       "      <td>0.996392</td>\n",
       "      <td>0.932610</td>\n",
       "      <td>17.597547</td>\n",
       "      <td>0.995547</td>\n",
       "      <td>0.927965</td>\n",
       "      <td>11.978888</td>\n",
       "      <td>0.995145</td>\n",
       "      <td>0.930084</td>\n",
       "      <td>16.360045</td>\n",
       "      <td>0.996540</td>\n",
       "      <td>0.929433</td>\n",
       "      <td>13.759318</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>MS</td>\n",
       "      <td>0.996307</td>\n",
       "      <td>0.930577</td>\n",
       "      <td>17.614616</td>\n",
       "      <td>0.996092</td>\n",
       "      <td>0.928760</td>\n",
       "      <td>14.499343</td>\n",
       "      <td>0.996527</td>\n",
       "      <td>0.930348</td>\n",
       "      <td>19.074713</td>\n",
       "      <td>0.996573</td>\n",
       "      <td>0.928718</td>\n",
       "      <td>18.022715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>MS</td>\n",
       "      <td>0.996981</td>\n",
       "      <td>0.929295</td>\n",
       "      <td>17.214503</td>\n",
       "      <td>0.996256</td>\n",
       "      <td>0.926172</td>\n",
       "      <td>10.464970</td>\n",
       "      <td>0.996594</td>\n",
       "      <td>0.928684</td>\n",
       "      <td>13.447240</td>\n",
       "      <td>0.996566</td>\n",
       "      <td>0.929728</td>\n",
       "      <td>15.503118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>MS</td>\n",
       "      <td>0.996710</td>\n",
       "      <td>0.922257</td>\n",
       "      <td>14.359057</td>\n",
       "      <td>0.995961</td>\n",
       "      <td>0.930695</td>\n",
       "      <td>21.827001</td>\n",
       "      <td>0.996219</td>\n",
       "      <td>0.928476</td>\n",
       "      <td>16.014446</td>\n",
       "      <td>0.996145</td>\n",
       "      <td>0.929053</td>\n",
       "      <td>14.34867</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>MS</td>\n",
       "      <td>0.996754</td>\n",
       "      <td>0.932096</td>\n",
       "      <td>22.130655</td>\n",
       "      <td>0.996055</td>\n",
       "      <td>0.931234</td>\n",
       "      <td>21.268408</td>\n",
       "      <td>0.996295</td>\n",
       "      <td>0.931816</td>\n",
       "      <td>21.958528</td>\n",
       "      <td>0.996179</td>\n",
       "      <td>0.930957</td>\n",
       "      <td>17.721097</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>MS</td>\n",
       "      <td>0.996639</td>\n",
       "      <td>0.931700</td>\n",
       "      <td>20.930679</td>\n",
       "      <td>0.996163</td>\n",
       "      <td>0.932225</td>\n",
       "      <td>25.893131</td>\n",
       "      <td>0.996298</td>\n",
       "      <td>0.927625</td>\n",
       "      <td>13.011789</td>\n",
       "      <td>0.996193</td>\n",
       "      <td>0.930103</td>\n",
       "      <td>16.548737</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>MS</td>\n",
       "      <td>0.996683</td>\n",
       "      <td>0.928091</td>\n",
       "      <td>12.334372</td>\n",
       "      <td>0.996116</td>\n",
       "      <td>0.932101</td>\n",
       "      <td>24.070397</td>\n",
       "      <td>0.996206</td>\n",
       "      <td>0.930190</td>\n",
       "      <td>17.322049</td>\n",
       "      <td>0.996479</td>\n",
       "      <td>0.928764</td>\n",
       "      <td>14.777348</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>MS</td>\n",
       "      <td>0.996612</td>\n",
       "      <td>0.928926</td>\n",
       "      <td>16.109173</td>\n",
       "      <td>0.996021</td>\n",
       "      <td>0.926278</td>\n",
       "      <td>10.361054</td>\n",
       "      <td>0.996066</td>\n",
       "      <td>0.930444</td>\n",
       "      <td>18.788448</td>\n",
       "      <td>0.996112</td>\n",
       "      <td>0.928960</td>\n",
       "      <td>15.959209</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>MS</td>\n",
       "      <td>0.992730</td>\n",
       "      <td>0.931406</td>\n",
       "      <td>21.083396</td>\n",
       "      <td>0.992964</td>\n",
       "      <td>0.929582</td>\n",
       "      <td>17.804908</td>\n",
       "      <td>0.994687</td>\n",
       "      <td>0.925624</td>\n",
       "      <td>13.022314</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>MS</td>\n",
       "      <td>0.995303</td>\n",
       "      <td>0.929025</td>\n",
       "      <td>17.604515</td>\n",
       "      <td>0.992786</td>\n",
       "      <td>0.929825</td>\n",
       "      <td>17.913527</td>\n",
       "      <td>0.994813</td>\n",
       "      <td>0.927596</td>\n",
       "      <td>14.005527</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>MS</td>\n",
       "      <td>0.992636</td>\n",
       "      <td>0.930224</td>\n",
       "      <td>18.606838</td>\n",
       "      <td>0.991778</td>\n",
       "      <td>0.924419</td>\n",
       "      <td>13.320239</td>\n",
       "      <td>0.995546</td>\n",
       "      <td>0.928420</td>\n",
       "      <td>16.123715</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>MS</td>\n",
       "      <td>0.992662</td>\n",
       "      <td>0.931293</td>\n",
       "      <td>21.928787</td>\n",
       "      <td>0.993317</td>\n",
       "      <td>0.930030</td>\n",
       "      <td>17.789350</td>\n",
       "      <td>0.995349</td>\n",
       "      <td>0.928532</td>\n",
       "      <td>15.572215</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>MS</td>\n",
       "      <td>0.992985</td>\n",
       "      <td>0.927435</td>\n",
       "      <td>12.558216</td>\n",
       "      <td>0.993369</td>\n",
       "      <td>0.930563</td>\n",
       "      <td>20.125712</td>\n",
       "      <td>0.995572</td>\n",
       "      <td>0.928229</td>\n",
       "      <td>14.886060</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unnamed: 1  Bisulfite conversion rate  Coverage      Depth  \\\n",
       "1          HC                   0.996512  0.930774  12.938460   \n",
       "2          HC                   0.996372  0.930213  17.840417   \n",
       "3          HC                   0.996646  0.930228  17.773436   \n",
       "4          HC                   0.996599  0.930274  18.368304   \n",
       "5          HC                   0.996699  0.929747  17.377918   \n",
       "6          HC                   0.996768  0.932192  23.408974   \n",
       "7          HC                   0.996947  0.929685  15.971407   \n",
       "8          HC                   0.996898  0.928607  13.548486   \n",
       "9          HC                   0.996820  0.930086  16.368358   \n",
       "10         HC                   0.996613  0.931072  19.442724   \n",
       "11         HC                   0.992790  0.931186  21.812578   \n",
       "12         HC                   0.992276  0.930920  20.684043   \n",
       "13         MS                   0.996657  0.930324  17.984730   \n",
       "14         MS                   0.996421  0.930266  17.007209   \n",
       "15         MS                   0.996751  0.929358  15.815312   \n",
       "16         MS                   0.996621  0.929229  14.935193   \n",
       "17         MS                   0.996392  0.932610  17.597547   \n",
       "18         MS                   0.996307  0.930577  17.614616   \n",
       "19         MS                   0.996981  0.929295  17.214503   \n",
       "20         MS                   0.996710  0.922257  14.359057   \n",
       "21         MS                   0.996754  0.932096  22.130655   \n",
       "22         MS                   0.996639  0.931700  20.930679   \n",
       "23         MS                   0.996683  0.928091  12.334372   \n",
       "24         MS                   0.996612  0.928926  16.109173   \n",
       "25         MS                   0.992730  0.931406  21.083396   \n",
       "26         MS                   0.995303  0.929025  17.604515   \n",
       "27         MS                   0.992636  0.930224  18.606838   \n",
       "28         MS                   0.992662  0.931293  21.928787   \n",
       "29         MS                   0.992985  0.927435  12.558216   \n",
       "\n",
       "    Bisulfite conversion rate.1  Coverage.1    Depth.1  \\\n",
       "1                      0.996427    0.929601  17.736396   \n",
       "2                      0.996214    0.929735  17.822934   \n",
       "3                      0.996185    0.928996  16.137003   \n",
       "4                      0.996369    0.926824  12.071238   \n",
       "5                      0.996125    0.929457  15.147257   \n",
       "6                      0.996068    0.928345  12.632632   \n",
       "7                      0.995900    0.925510   9.288985   \n",
       "8                      0.996175    0.926523  10.525559   \n",
       "9                      0.996251    0.927231  10.924250   \n",
       "10                     0.996034    0.931463  22.572802   \n",
       "11                     0.992614    0.929490  17.864039   \n",
       "12                     0.992878    0.930062  18.524907   \n",
       "13                     0.996089    0.929900  17.037665   \n",
       "14                     0.996215    0.927086  12.921820   \n",
       "15                     0.996331    0.929285  17.466062   \n",
       "16                     0.996349    0.928237  15.905936   \n",
       "17                     0.995547    0.927965  11.978888   \n",
       "18                     0.996092    0.928760  14.499343   \n",
       "19                     0.996256    0.926172  10.464970   \n",
       "20                     0.995961    0.930695  21.827001   \n",
       "21                     0.996055    0.931234  21.268408   \n",
       "22                     0.996163    0.932225  25.893131   \n",
       "23                     0.996116    0.932101  24.070397   \n",
       "24                     0.996021    0.926278  10.361054   \n",
       "25                     0.992964    0.929582  17.804908   \n",
       "26                     0.992786    0.929825  17.913527   \n",
       "27                     0.991778    0.924419  13.320239   \n",
       "28                     0.993317    0.930030  17.789350   \n",
       "29                     0.993369    0.930563  20.125712   \n",
       "\n",
       "    Bisulfite conversion rate.2  Coverage.2    Depth.2  \\\n",
       "1                      0.995716    0.929011  15.337275   \n",
       "2                      0.995485    0.929513  16.006528   \n",
       "3                      0.995200    0.930783  19.982329   \n",
       "4                      0.995258    0.929191  16.832175   \n",
       "5                      0.995298    0.929502  16.636607   \n",
       "6                      0.995264    0.930080  14.346573   \n",
       "7                      0.996674    0.929717  15.750571   \n",
       "8                      0.996554    0.929193  14.511543   \n",
       "9                      0.996321    0.930720  16.527435   \n",
       "10                     0.996316    0.930101  15.576498   \n",
       "11                     0.995146    0.930830  22.355377   \n",
       "12                     0.994130    0.930743  22.457742   \n",
       "13                     0.995197    0.929483  17.773411   \n",
       "14                     0.995429    0.928552  14.450575   \n",
       "15                     0.995302    0.929864  17.509609   \n",
       "16                     0.995421    0.928881  15.297445   \n",
       "17                     0.995145    0.930084  16.360045   \n",
       "18                     0.996527    0.930348  19.074713   \n",
       "19                     0.996594    0.928684  13.447240   \n",
       "20                     0.996219    0.928476  16.014446   \n",
       "21                     0.996295    0.931816  21.958528   \n",
       "22                     0.996298    0.927625  13.011789   \n",
       "23                     0.996206    0.930190  17.322049   \n",
       "24                     0.996066    0.930444  18.788448   \n",
       "25                     0.994687    0.925624  13.022314   \n",
       "26                     0.994813    0.927596  14.005527   \n",
       "27                     0.995546    0.928420  16.123715   \n",
       "28                     0.995349    0.928532  15.572215   \n",
       "29                     0.995572    0.928229  14.886060   \n",
       "\n",
       "    Bisulfite conversion rate.3  Coverage.3    Depth.3  \n",
       "1                      0.996680    0.927372  13.229731  \n",
       "2                      0.996716    0.929211  16.705444  \n",
       "3                      0.996675    0.929492  16.934506  \n",
       "4                      0.996455    0.928490  13.884959  \n",
       "5                      0.996767    0.929995  19.246635  \n",
       "6                      0.996661    0.928186  14.603209  \n",
       "7                      0.996651    0.929517  14.622796  \n",
       "8                      0.996471    0.927907  13.028801  \n",
       "9                      0.996473    0.930439  17.994847  \n",
       "10                     0.996176    0.926703  12.642628  \n",
       "11                          NaN         NaN        NaN  \n",
       "12                          NaN         NaN        NaN  \n",
       "13                     0.996676    0.928924  13.923285  \n",
       "14                     0.996626    0.930401  19.157674  \n",
       "15                     0.996735    0.928064  13.641882  \n",
       "16                     0.996731    0.928808  14.776338  \n",
       "17                     0.996540    0.929433  13.759318  \n",
       "18                     0.996573    0.928718  18.022715  \n",
       "19                     0.996566    0.929728  15.503118  \n",
       "20                     0.996145    0.929053   14.34867  \n",
       "21                     0.996179    0.930957  17.721097  \n",
       "22                     0.996193    0.930103  16.548737  \n",
       "23                     0.996479    0.928764  14.777348  \n",
       "24                     0.996112    0.928960  15.959209  \n",
       "25                          NaN         NaN        NaN  \n",
       "26                          NaN         NaN         NA  \n",
       "27                          NaN         NaN        NaN  \n",
       "28                          NaN         NaN        NaN  \n",
       "29                          NaN         NaN        NaN  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_01 = pd.read_excel('pnas.2111920118.sd01.xlsx', header=1, index_col=0)\n",
    "data_01"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample ID</th>\n",
       "      <th>Group</th>\n",
       "      <th>Bisulfite conversion rate</th>\n",
       "      <th>Coverage</th>\n",
       "      <th>Depth</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HC-1</td>\n",
       "      <td>HC</td>\n",
       "      <td>0.995716</td>\n",
       "      <td>0.929206</td>\n",
       "      <td>15.736371</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HC-2</td>\n",
       "      <td>HC</td>\n",
       "      <td>0.995485</td>\n",
       "      <td>0.929513</td>\n",
       "      <td>16.006528</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HC-3</td>\n",
       "      <td>HC</td>\n",
       "      <td>0.995200</td>\n",
       "      <td>0.930783</td>\n",
       "      <td>19.982329</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HC-4</td>\n",
       "      <td>HC</td>\n",
       "      <td>0.995258</td>\n",
       "      <td>0.929191</td>\n",
       "      <td>16.832175</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HC-5</td>\n",
       "      <td>HC</td>\n",
       "      <td>0.995298</td>\n",
       "      <td>0.929502</td>\n",
       "      <td>16.636607</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>HC-6</td>\n",
       "      <td>HC</td>\n",
       "      <td>0.995264</td>\n",
       "      <td>0.930080</td>\n",
       "      <td>14.346573</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HC-7</td>\n",
       "      <td>HC</td>\n",
       "      <td>0.996674</td>\n",
       "      <td>0.929717</td>\n",
       "      <td>15.750571</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>HC-8</td>\n",
       "      <td>HC</td>\n",
       "      <td>0.996554</td>\n",
       "      <td>0.929193</td>\n",
       "      <td>14.511543</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>HC-9</td>\n",
       "      <td>HC</td>\n",
       "      <td>0.996321</td>\n",
       "      <td>0.930720</td>\n",
       "      <td>16.527435</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>HC-10</td>\n",
       "      <td>HC</td>\n",
       "      <td>0.996316</td>\n",
       "      <td>0.930101</td>\n",
       "      <td>15.576498</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>HC-11</td>\n",
       "      <td>HC</td>\n",
       "      <td>0.995146</td>\n",
       "      <td>0.930830</td>\n",
       "      <td>22.355377</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>HC-12</td>\n",
       "      <td>HC</td>\n",
       "      <td>0.994130</td>\n",
       "      <td>0.930743</td>\n",
       "      <td>22.457742</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HC-13</td>\n",
       "      <td>HC</td>\n",
       "      <td>0.992355</td>\n",
       "      <td>0.926394</td>\n",
       "      <td>10.391535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>HC-14</td>\n",
       "      <td>HC</td>\n",
       "      <td>0.992748</td>\n",
       "      <td>0.926473</td>\n",
       "      <td>10.187587</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>HC-15</td>\n",
       "      <td>HC</td>\n",
       "      <td>0.994182</td>\n",
       "      <td>0.925834</td>\n",
       "      <td>9.406651</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>HC-16</td>\n",
       "      <td>HC</td>\n",
       "      <td>0.995444</td>\n",
       "      <td>0.927235</td>\n",
       "      <td>11.569384</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>HC-17</td>\n",
       "      <td>HC</td>\n",
       "      <td>0.995469</td>\n",
       "      <td>0.926792</td>\n",
       "      <td>10.560540</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>HC-18</td>\n",
       "      <td>HC</td>\n",
       "      <td>0.996137</td>\n",
       "      <td>0.927140</td>\n",
       "      <td>10.581697</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>HC-19</td>\n",
       "      <td>HC</td>\n",
       "      <td>0.995572</td>\n",
       "      <td>0.929846</td>\n",
       "      <td>16.535056</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>HC-20</td>\n",
       "      <td>HC</td>\n",
       "      <td>0.995508</td>\n",
       "      <td>0.930034</td>\n",
       "      <td>15.527802</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>HC-21</td>\n",
       "      <td>HC</td>\n",
       "      <td>0.995096</td>\n",
       "      <td>0.929774</td>\n",
       "      <td>15.519757</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>HC-22</td>\n",
       "      <td>HC</td>\n",
       "      <td>0.994387</td>\n",
       "      <td>0.930242</td>\n",
       "      <td>16.043092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>HC-23</td>\n",
       "      <td>HC</td>\n",
       "      <td>0.994635</td>\n",
       "      <td>0.930027</td>\n",
       "      <td>15.563786</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>HC-24</td>\n",
       "      <td>HC</td>\n",
       "      <td>0.994710</td>\n",
       "      <td>0.929430</td>\n",
       "      <td>14.787978</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>MS-1</td>\n",
       "      <td>MS</td>\n",
       "      <td>0.995197</td>\n",
       "      <td>0.929483</td>\n",
       "      <td>17.773411</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>MS-2</td>\n",
       "      <td>MS</td>\n",
       "      <td>0.995429</td>\n",
       "      <td>0.928552</td>\n",
       "      <td>14.450575</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>MS-3</td>\n",
       "      <td>MS</td>\n",
       "      <td>0.995302</td>\n",
       "      <td>0.929864</td>\n",
       "      <td>17.509609</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>MS-4</td>\n",
       "      <td>MS</td>\n",
       "      <td>0.995419</td>\n",
       "      <td>0.929271</td>\n",
       "      <td>16.134690</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>MS-5</td>\n",
       "      <td>MS</td>\n",
       "      <td>0.995145</td>\n",
       "      <td>0.930084</td>\n",
       "      <td>16.360045</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>MS-6</td>\n",
       "      <td>MS</td>\n",
       "      <td>0.996527</td>\n",
       "      <td>0.930348</td>\n",
       "      <td>19.074713</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>MS-7</td>\n",
       "      <td>MS</td>\n",
       "      <td>0.996594</td>\n",
       "      <td>0.928684</td>\n",
       "      <td>13.447240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>MS-8</td>\n",
       "      <td>MS</td>\n",
       "      <td>0.996219</td>\n",
       "      <td>0.928476</td>\n",
       "      <td>16.014446</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>MS-9</td>\n",
       "      <td>MS</td>\n",
       "      <td>0.996295</td>\n",
       "      <td>0.931816</td>\n",
       "      <td>21.958528</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>MS-10</td>\n",
       "      <td>MS</td>\n",
       "      <td>0.996298</td>\n",
       "      <td>0.927625</td>\n",
       "      <td>13.011789</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>MS-11</td>\n",
       "      <td>MS</td>\n",
       "      <td>0.996206</td>\n",
       "      <td>0.930190</td>\n",
       "      <td>17.322049</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>MS-12</td>\n",
       "      <td>MS</td>\n",
       "      <td>0.996066</td>\n",
       "      <td>0.930444</td>\n",
       "      <td>18.788448</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>MS-13</td>\n",
       "      <td>MS</td>\n",
       "      <td>0.994687</td>\n",
       "      <td>0.925624</td>\n",
       "      <td>13.022314</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>MS-14</td>\n",
       "      <td>MS</td>\n",
       "      <td>0.994813</td>\n",
       "      <td>0.927596</td>\n",
       "      <td>14.005527</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>MS-15</td>\n",
       "      <td>MS</td>\n",
       "      <td>0.995546</td>\n",
       "      <td>0.928420</td>\n",
       "      <td>16.123715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>MS-16</td>\n",
       "      <td>MS</td>\n",
       "      <td>0.995349</td>\n",
       "      <td>0.928532</td>\n",
       "      <td>15.572215</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>MS-17</td>\n",
       "      <td>MS</td>\n",
       "      <td>0.995572</td>\n",
       "      <td>0.928229</td>\n",
       "      <td>14.886060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>MS-18</td>\n",
       "      <td>MS</td>\n",
       "      <td>0.995828</td>\n",
       "      <td>0.926327</td>\n",
       "      <td>10.210302</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>MS-19</td>\n",
       "      <td>MS</td>\n",
       "      <td>0.996110</td>\n",
       "      <td>0.927709</td>\n",
       "      <td>12.458151</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>MS-20</td>\n",
       "      <td>MS</td>\n",
       "      <td>0.995342</td>\n",
       "      <td>0.926395</td>\n",
       "      <td>10.736239</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>MS-21</td>\n",
       "      <td>MS</td>\n",
       "      <td>0.995563</td>\n",
       "      <td>0.927048</td>\n",
       "      <td>11.156173</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>MS-22</td>\n",
       "      <td>MS</td>\n",
       "      <td>0.995760</td>\n",
       "      <td>0.928579</td>\n",
       "      <td>12.637373</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>MS-23</td>\n",
       "      <td>MS</td>\n",
       "      <td>0.995999</td>\n",
       "      <td>0.927195</td>\n",
       "      <td>10.727717</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>MS-24</td>\n",
       "      <td>MS</td>\n",
       "      <td>0.995630</td>\n",
       "      <td>0.929874</td>\n",
       "      <td>14.777576</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>MS-25</td>\n",
       "      <td>MS</td>\n",
       "      <td>0.995670</td>\n",
       "      <td>0.930385</td>\n",
       "      <td>16.597375</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>MS-26</td>\n",
       "      <td>MS</td>\n",
       "      <td>0.994624</td>\n",
       "      <td>0.930171</td>\n",
       "      <td>15.797287</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>MS-27</td>\n",
       "      <td>MS</td>\n",
       "      <td>0.994782</td>\n",
       "      <td>0.930741</td>\n",
       "      <td>17.314463</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>MS-28</td>\n",
       "      <td>MS</td>\n",
       "      <td>0.994965</td>\n",
       "      <td>0.929931</td>\n",
       "      <td>15.496770</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>MS-29</td>\n",
       "      <td>MS</td>\n",
       "      <td>0.994669</td>\n",
       "      <td>0.929593</td>\n",
       "      <td>15.578220</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Sample ID Group  Bisulfite conversion rate  Coverage      Depth\n",
       "0       HC-1    HC                   0.995716  0.929206  15.736371\n",
       "1       HC-2    HC                   0.995485  0.929513  16.006528\n",
       "2       HC-3    HC                   0.995200  0.930783  19.982329\n",
       "3       HC-4    HC                   0.995258  0.929191  16.832175\n",
       "4       HC-5    HC                   0.995298  0.929502  16.636607\n",
       "5       HC-6    HC                   0.995264  0.930080  14.346573\n",
       "6       HC-7    HC                   0.996674  0.929717  15.750571\n",
       "7       HC-8    HC                   0.996554  0.929193  14.511543\n",
       "8       HC-9    HC                   0.996321  0.930720  16.527435\n",
       "9      HC-10    HC                   0.996316  0.930101  15.576498\n",
       "10     HC-11    HC                   0.995146  0.930830  22.355377\n",
       "11     HC-12    HC                   0.994130  0.930743  22.457742\n",
       "12     HC-13    HC                   0.992355  0.926394  10.391535\n",
       "13     HC-14    HC                   0.992748  0.926473  10.187587\n",
       "14     HC-15    HC                   0.994182  0.925834   9.406651\n",
       "15     HC-16    HC                   0.995444  0.927235  11.569384\n",
       "16     HC-17    HC                   0.995469  0.926792  10.560540\n",
       "17     HC-18    HC                   0.996137  0.927140  10.581697\n",
       "18     HC-19    HC                   0.995572  0.929846  16.535056\n",
       "19     HC-20    HC                   0.995508  0.930034  15.527802\n",
       "20     HC-21    HC                   0.995096  0.929774  15.519757\n",
       "21     HC-22    HC                   0.994387  0.930242  16.043092\n",
       "22     HC-23    HC                   0.994635  0.930027  15.563786\n",
       "23     HC-24    HC                   0.994710  0.929430  14.787978\n",
       "24      MS-1    MS                   0.995197  0.929483  17.773411\n",
       "25      MS-2    MS                   0.995429  0.928552  14.450575\n",
       "26      MS-3    MS                   0.995302  0.929864  17.509609\n",
       "27      MS-4    MS                   0.995419  0.929271  16.134690\n",
       "28      MS-5    MS                   0.995145  0.930084  16.360045\n",
       "29      MS-6    MS                   0.996527  0.930348  19.074713\n",
       "30      MS-7    MS                   0.996594  0.928684  13.447240\n",
       "31      MS-8    MS                   0.996219  0.928476  16.014446\n",
       "32      MS-9    MS                   0.996295  0.931816  21.958528\n",
       "33     MS-10    MS                   0.996298  0.927625  13.011789\n",
       "34     MS-11    MS                   0.996206  0.930190  17.322049\n",
       "35     MS-12    MS                   0.996066  0.930444  18.788448\n",
       "36     MS-13    MS                   0.994687  0.925624  13.022314\n",
       "37     MS-14    MS                   0.994813  0.927596  14.005527\n",
       "38     MS-15    MS                   0.995546  0.928420  16.123715\n",
       "39     MS-16    MS                   0.995349  0.928532  15.572215\n",
       "40     MS-17    MS                   0.995572  0.928229  14.886060\n",
       "41     MS-18    MS                   0.995828  0.926327  10.210302\n",
       "42     MS-19    MS                   0.996110  0.927709  12.458151\n",
       "43     MS-20    MS                   0.995342  0.926395  10.736239\n",
       "44     MS-21    MS                   0.995563  0.927048  11.156173\n",
       "45     MS-22    MS                   0.995760  0.928579  12.637373\n",
       "46     MS-23    MS                   0.995999  0.927195  10.727717\n",
       "47     MS-24    MS                   0.995630  0.929874  14.777576\n",
       "48     MS-25    MS                   0.995670  0.930385  16.597375\n",
       "49     MS-26    MS                   0.994624  0.930171  15.797287\n",
       "50     MS-27    MS                   0.994782  0.930741  17.314463\n",
       "51     MS-28    MS                   0.994965  0.929931  15.496770\n",
       "52     MS-29    MS                   0.994669  0.929593  15.578220"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_02 = pd.read_excel('pnas.2111920118.sd02.xlsx')\n",
    "data_02"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample ID</th>\n",
       "      <th>Group</th>\n",
       "      <th>Dataset</th>\n",
       "      <th>Age</th>\n",
       "      <th>Gender</th>\n",
       "      <th>EDSS</th>\n",
       "      <th>Brain lesion burden</th>\n",
       "      <th>Brain Gd</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HC-1</td>\n",
       "      <td>HC</td>\n",
       "      <td>1.0</td>\n",
       "      <td>58.0</td>\n",
       "      <td>F</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>HC-2</td>\n",
       "      <td>HC</td>\n",
       "      <td>1.0</td>\n",
       "      <td>58.0</td>\n",
       "      <td>F</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HC-3</td>\n",
       "      <td>HC</td>\n",
       "      <td>1.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>F</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>HC-4</td>\n",
       "      <td>HC</td>\n",
       "      <td>1.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>F</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HC-5</td>\n",
       "      <td>HC</td>\n",
       "      <td>1.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>F</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>HC-6</td>\n",
       "      <td>HC</td>\n",
       "      <td>1.0</td>\n",
       "      <td>54.0</td>\n",
       "      <td>F</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>HC-7</td>\n",
       "      <td>HC</td>\n",
       "      <td>1.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>F</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>HC-8</td>\n",
       "      <td>HC</td>\n",
       "      <td>1.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>F</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>HC-9</td>\n",
       "      <td>HC</td>\n",
       "      <td>1.0</td>\n",
       "      <td>58.0</td>\n",
       "      <td>F</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>HC-10</td>\n",
       "      <td>HC</td>\n",
       "      <td>1.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>F</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>HC-11</td>\n",
       "      <td>HC</td>\n",
       "      <td>1.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>F</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>HC-12</td>\n",
       "      <td>HC</td>\n",
       "      <td>1.0</td>\n",
       "      <td>22.0</td>\n",
       "      <td>F</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HC-13</td>\n",
       "      <td>HC</td>\n",
       "      <td>2.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>F</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>HC-14</td>\n",
       "      <td>HC</td>\n",
       "      <td>2.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>F</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>HC-15</td>\n",
       "      <td>HC</td>\n",
       "      <td>2.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>F</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>HC-16</td>\n",
       "      <td>HC</td>\n",
       "      <td>2.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>F</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>HC-17</td>\n",
       "      <td>HC</td>\n",
       "      <td>2.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>F</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>HC-18</td>\n",
       "      <td>HC</td>\n",
       "      <td>2.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>F</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>HC-19</td>\n",
       "      <td>HC</td>\n",
       "      <td>2.0</td>\n",
       "      <td>39.0</td>\n",
       "      <td>F</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>HC-20</td>\n",
       "      <td>HC</td>\n",
       "      <td>2.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>F</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>HC-21</td>\n",
       "      <td>HC</td>\n",
       "      <td>2.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>F</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>HC-22</td>\n",
       "      <td>HC</td>\n",
       "      <td>2.0</td>\n",
       "      <td>28.0</td>\n",
       "      <td>F</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>HC-23</td>\n",
       "      <td>HC</td>\n",
       "      <td>2.0</td>\n",
       "      <td>45.0</td>\n",
       "      <td>F</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>HC-24</td>\n",
       "      <td>HC</td>\n",
       "      <td>2.0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>F</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>MS-1</td>\n",
       "      <td>MS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>56.0</td>\n",
       "      <td>F</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>MS-2</td>\n",
       "      <td>MS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>43.0</td>\n",
       "      <td>F</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>MS-3</td>\n",
       "      <td>MS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>F</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>MS-4</td>\n",
       "      <td>MS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>54.0</td>\n",
       "      <td>F</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>MS-5</td>\n",
       "      <td>MS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>42.0</td>\n",
       "      <td>F</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>MS-6</td>\n",
       "      <td>MS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>43.0</td>\n",
       "      <td>F</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>MS-7</td>\n",
       "      <td>MS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>39.0</td>\n",
       "      <td>F</td>\n",
       "      <td>1.5</td>\n",
       "      <td>3.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>MS-8</td>\n",
       "      <td>MS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>F</td>\n",
       "      <td>1.5</td>\n",
       "      <td>3.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>MS-9</td>\n",
       "      <td>MS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>33.0</td>\n",
       "      <td>F</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>MS-10</td>\n",
       "      <td>MS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>F</td>\n",
       "      <td>1.5</td>\n",
       "      <td>2.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>MS-11</td>\n",
       "      <td>MS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>41.0</td>\n",
       "      <td>F</td>\n",
       "      <td>2.5</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>MS-12</td>\n",
       "      <td>MS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>F</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>MS-13</td>\n",
       "      <td>MS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>F</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>MS-14</td>\n",
       "      <td>MS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>F</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>MS-15</td>\n",
       "      <td>MS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>F</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>MS-16</td>\n",
       "      <td>MS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>F</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>MS-17</td>\n",
       "      <td>MS</td>\n",
       "      <td>1.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>F</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>MS-18</td>\n",
       "      <td>MS</td>\n",
       "      <td>2.0</td>\n",
       "      <td>55.0</td>\n",
       "      <td>F</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>MS-19</td>\n",
       "      <td>MS</td>\n",
       "      <td>2.0</td>\n",
       "      <td>32.0</td>\n",
       "      <td>F</td>\n",
       "      <td>1.5</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>MS-20</td>\n",
       "      <td>MS</td>\n",
       "      <td>2.0</td>\n",
       "      <td>30.0</td>\n",
       "      <td>F</td>\n",
       "      <td>2.5</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>MS-21</td>\n",
       "      <td>MS</td>\n",
       "      <td>2.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>F</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>MS-22</td>\n",
       "      <td>MS</td>\n",
       "      <td>2.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>F</td>\n",
       "      <td>1.5</td>\n",
       "      <td>2.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>MS-23</td>\n",
       "      <td>MS</td>\n",
       "      <td>2.0</td>\n",
       "      <td>31.0</td>\n",
       "      <td>F</td>\n",
       "      <td>3.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>MS-24</td>\n",
       "      <td>MS</td>\n",
       "      <td>2.0</td>\n",
       "      <td>25.0</td>\n",
       "      <td>F</td>\n",
       "      <td>1.5</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>MS-25</td>\n",
       "      <td>MS</td>\n",
       "      <td>2.0</td>\n",
       "      <td>44.0</td>\n",
       "      <td>F</td>\n",
       "      <td>1.5</td>\n",
       "      <td>2.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>MS-26</td>\n",
       "      <td>MS</td>\n",
       "      <td>2.0</td>\n",
       "      <td>37.0</td>\n",
       "      <td>F</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>MS-27</td>\n",
       "      <td>MS</td>\n",
       "      <td>2.0</td>\n",
       "      <td>35.0</td>\n",
       "      <td>F</td>\n",
       "      <td>2.5</td>\n",
       "      <td>2.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>MS-28</td>\n",
       "      <td>MS</td>\n",
       "      <td>2.0</td>\n",
       "      <td>48.0</td>\n",
       "      <td>F</td>\n",
       "      <td>2.5</td>\n",
       "      <td>1.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>MS-29</td>\n",
       "      <td>MS</td>\n",
       "      <td>2.0</td>\n",
       "      <td>38.0</td>\n",
       "      <td>F</td>\n",
       "      <td>3.5</td>\n",
       "      <td>3.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>Brain lesion burdern: 0 = 0 lesions; 1 = 0-9 l...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                            Sample ID Group  Dataset   Age  \\\n",
       "0                                                HC-1    HC      1.0  58.0   \n",
       "1                                                HC-2    HC      1.0  58.0   \n",
       "2                                                HC-3    HC      1.0  31.0   \n",
       "3                                                HC-4    HC      1.0  31.0   \n",
       "4                                                HC-5    HC      1.0  25.0   \n",
       "5                                                HC-6    HC      1.0  54.0   \n",
       "6                                                HC-7    HC      1.0  23.0   \n",
       "7                                                HC-8    HC      1.0  50.0   \n",
       "8                                                HC-9    HC      1.0  58.0   \n",
       "9                                               HC-10    HC      1.0  33.0   \n",
       "10                                              HC-11    HC      1.0  33.0   \n",
       "11                                              HC-12    HC      1.0  22.0   \n",
       "12                                              HC-13    HC      2.0  46.0   \n",
       "13                                              HC-14    HC      2.0  27.0   \n",
       "14                                              HC-15    HC      2.0  32.0   \n",
       "15                                              HC-16    HC      2.0  27.0   \n",
       "16                                              HC-17    HC      2.0  40.0   \n",
       "17                                              HC-18    HC      2.0  27.0   \n",
       "18                                              HC-19    HC      2.0  39.0   \n",
       "19                                              HC-20    HC      2.0  25.0   \n",
       "20                                              HC-21    HC      2.0  28.0   \n",
       "21                                              HC-22    HC      2.0  28.0   \n",
       "22                                              HC-23    HC      2.0  45.0   \n",
       "23                                              HC-24    HC      2.0  47.0   \n",
       "24                                               MS-1    MS      1.0  56.0   \n",
       "25                                               MS-2    MS      1.0  43.0   \n",
       "26                                               MS-3    MS      1.0  36.0   \n",
       "27                                               MS-4    MS      1.0  54.0   \n",
       "28                                               MS-5    MS      1.0  42.0   \n",
       "29                                               MS-6    MS      1.0  43.0   \n",
       "30                                               MS-7    MS      1.0  39.0   \n",
       "31                                               MS-8    MS      1.0  35.0   \n",
       "32                                               MS-9    MS      1.0  33.0   \n",
       "33                                              MS-10    MS      1.0  36.0   \n",
       "34                                              MS-11    MS      1.0  41.0   \n",
       "35                                              MS-12    MS      1.0  20.0   \n",
       "36                                              MS-13    MS      1.0  36.0   \n",
       "37                                              MS-14    MS      1.0  40.0   \n",
       "38                                              MS-15    MS      1.0  25.0   \n",
       "39                                              MS-16    MS      1.0  32.0   \n",
       "40                                              MS-17    MS      1.0  50.0   \n",
       "41                                              MS-18    MS      2.0  55.0   \n",
       "42                                              MS-19    MS      2.0  32.0   \n",
       "43                                              MS-20    MS      2.0  30.0   \n",
       "44                                              MS-21    MS      2.0  49.0   \n",
       "45                                              MS-22    MS      2.0  21.0   \n",
       "46                                              MS-23    MS      2.0  31.0   \n",
       "47                                              MS-24    MS      2.0  25.0   \n",
       "48                                              MS-25    MS      2.0  44.0   \n",
       "49                                              MS-26    MS      2.0  37.0   \n",
       "50                                              MS-27    MS      2.0  35.0   \n",
       "51                                              MS-28    MS      2.0  48.0   \n",
       "52                                              MS-29    MS      2.0  38.0   \n",
       "53                                                NaN   NaN      NaN   NaN   \n",
       "54                                                NaN   NaN      NaN   NaN   \n",
       "55  Brain lesion burdern: 0 = 0 lesions; 1 = 0-9 l...   NaN      NaN   NaN   \n",
       "\n",
       "   Gender  EDSS  Brain lesion burden Brain Gd  \n",
       "0       F   NaN                  NaN      NaN  \n",
       "1       F   NaN                  NaN      NaN  \n",
       "2       F   NaN                  NaN      NaN  \n",
       "3       F   NaN                  NaN      NaN  \n",
       "4       F   NaN                  NaN      NaN  \n",
       "5       F   NaN                  NaN      NaN  \n",
       "6       F   NaN                  NaN      NaN  \n",
       "7       F   NaN                  NaN      NaN  \n",
       "8       F   NaN                  NaN      NaN  \n",
       "9       F   NaN                  NaN      NaN  \n",
       "10      F   NaN                  NaN      NaN  \n",
       "11      F   NaN                  NaN      NaN  \n",
       "12      F   NaN                  NaN      NaN  \n",
       "13      F   NaN                  NaN      NaN  \n",
       "14      F   NaN                  NaN      NaN  \n",
       "15      F   NaN                  NaN      NaN  \n",
       "16      F   NaN                  NaN      NaN  \n",
       "17      F   NaN                  NaN      NaN  \n",
       "18      F   NaN                  NaN      NaN  \n",
       "19      F   NaN                  NaN      NaN  \n",
       "20      F   NaN                  NaN      NaN  \n",
       "21      F   NaN                  NaN      NaN  \n",
       "22      F   NaN                  NaN      NaN  \n",
       "23      F   NaN                  NaN      NaN  \n",
       "24      F   2.0                  2.0        -  \n",
       "25      F   2.0                  2.0        -  \n",
       "26      F   2.0                  1.0        -  \n",
       "27      F   2.0                  2.0        +  \n",
       "28      F   3.0                  2.0        -  \n",
       "29      F   3.0                  2.0        -  \n",
       "30      F   1.5                  3.0        +  \n",
       "31      F   1.5                  3.0        +  \n",
       "32      F   1.0                  2.0        +  \n",
       "33      F   1.5                  2.0        +  \n",
       "34      F   2.5                  2.0        -  \n",
       "35      F   2.0                  3.0        +  \n",
       "36      F   2.0                  1.0        -  \n",
       "37      F   3.0                  1.0        -  \n",
       "38      F   2.0                  2.0        -  \n",
       "39      F   1.0                  3.0        +  \n",
       "40      F   4.0                  1.0        -  \n",
       "41      F   2.0                  2.0        -  \n",
       "42      F   1.5                  1.0        -  \n",
       "43      F   2.5                  1.0        -  \n",
       "44      F   2.0                  2.0        -  \n",
       "45      F   1.5                  2.0        +  \n",
       "46      F   3.5                  NaN      NaN  \n",
       "47      F   1.5                  1.0        -  \n",
       "48      F   1.5                  2.0        +  \n",
       "49      F   2.0                  1.0        -  \n",
       "50      F   2.5                  2.0        +  \n",
       "51      F   2.5                  1.0        +  \n",
       "52      F   3.5                  3.0        +  \n",
       "53    NaN   NaN                  NaN      NaN  \n",
       "54    NaN   NaN                  NaN      NaN  \n",
       "55    NaN   NaN                  NaN      NaN  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_03 = pd.read_excel('pnas.2111920118.sd03.xlsx')\n",
    "data_03"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hypo-DMR assigned genes</th>\n",
       "      <th>Hyper-DMR assigned genes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>A1BG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A4GALT</td>\n",
       "      <td>ABCC4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AACS</td>\n",
       "      <td>ABCG1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AADACL2</td>\n",
       "      <td>ADAM17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAK1</td>\n",
       "      <td>ADAMTS9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5252</th>\n",
       "      <td>ZSWIM6</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5253</th>\n",
       "      <td>ZWILCH</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5254</th>\n",
       "      <td>ZWINT</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5255</th>\n",
       "      <td>ZYG11A</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5256</th>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5257 rows  2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Hypo-DMR assigned genes Hyper-DMR assigned genes\n",
       "0                       A1BG                     A1BG\n",
       "1                     A4GALT                    ABCC4\n",
       "2                       AACS                    ABCG1\n",
       "3                    AADACL2                   ADAM17\n",
       "4                       AAK1                  ADAMTS9\n",
       "...                      ...                      ...\n",
       "5252                  ZSWIM6                      NaN\n",
       "5253                  ZWILCH                      NaN\n",
       "5254                   ZWINT                      NaN\n",
       "5255                  ZYG11A                      NaN\n",
       "5256                   ZZEF1                      NaN\n",
       "\n",
       "[5257 rows x 2 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_04 = pd.read_excel('pnas.2111920118.sd04.xlsx')\n",
    "data_04"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Transcription factors</th>\n",
       "      <th>P-value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BATF</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>JUNB</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ATF3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MEF2A</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>IRF3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>MEF2B</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>MEF2C</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>IRF4</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>USF2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>PAX5</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>MAFK</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>BACH1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>MAX</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>ATF7</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>PAX8</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>ZEB1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Transcription factors  P-value\n",
       "0                   BATF        0\n",
       "1                   JUNB        0\n",
       "2                   ATF3        0\n",
       "3                  MEF2A        0\n",
       "4                   IRF3        0\n",
       "5                  MEF2B        0\n",
       "6                  MEF2C        0\n",
       "7                   IRF4        0\n",
       "8                   USF2        0\n",
       "9                   PAX5        0\n",
       "10                  MAFK        0\n",
       "11                 BACH1        0\n",
       "12                   MAX        0\n",
       "13                  ATF7        0\n",
       "14                  PAX8        0\n",
       "15                  ZEB1        0"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_05 = pd.read_excel('pnas.2111920118.sd05.xlsx')\n",
    "data_05"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Factor</th>\n",
       "      <th>P-value</th>\n",
       "      <th>Unnamed: 2</th>\n",
       "      <th>Unnamed: 3</th>\n",
       "      <th>Unnamed: 4</th>\n",
       "      <th>Unnamed: 5</th>\n",
       "      <th>Unnamed: 6</th>\n",
       "      <th>Unnamed: 7</th>\n",
       "      <th>Unnamed: 8</th>\n",
       "      <th>Unnamed: 9</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>up2k</td>\n",
       "      <td>up1.5k</td>\n",
       "      <td>up1k</td>\n",
       "      <td>up0.5k</td>\n",
       "      <td>hyper-DMRs</td>\n",
       "      <td>down0.5k</td>\n",
       "      <td>down1k</td>\n",
       "      <td>down1.5k</td>\n",
       "      <td>down2k</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>EWS:ERG</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.001</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>HOXC9</td>\n",
       "      <td>0.1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.001</td>\n",
       "      <td>0.01</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SCL</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.001</td>\n",
       "      <td>1</td>\n",
       "      <td>0.1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ASCL1</td>\n",
       "      <td>0.1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.01</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ATOH1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.01</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>CMYC</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.01</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>DUX</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.01</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ER71</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.01</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ERG</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.01</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ETS1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.01</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ETV1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.01</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>EWS:FLI1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.01</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>FLI</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.01</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>MYF5</td>\n",
       "      <td>0.1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.01</td>\n",
       "      <td>1</td>\n",
       "      <td>0.1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>MYOD</td>\n",
       "      <td>0.1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.01</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>MYOG</td>\n",
       "      <td>0.1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.01</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>PAX7</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.01</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>POLII</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.01</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>PPARG</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.01</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>RUNX2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.01</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>TCF12</td>\n",
       "      <td>0.01</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.01</td>\n",
       "      <td>1</td>\n",
       "      <td>0.1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>TWIST2</td>\n",
       "      <td>0.1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.01</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Factor P-value Unnamed: 2 Unnamed: 3 Unnamed: 4  Unnamed: 5 Unnamed: 6  \\\n",
       "0        NaN    up2k     up1.5k       up1k     up0.5k  hyper-DMRs   down0.5k   \n",
       "1    EWS:ERG       1          1          1          1       0.001          1   \n",
       "2      HOXC9     0.1          1          1          1       0.001       0.01   \n",
       "3        SCL       1          1          1          1       0.001          1   \n",
       "4      ASCL1     0.1          1          1          1        0.01          1   \n",
       "5      ATOH1       1          1          1          1        0.01          1   \n",
       "6       CMYC       1          1          1        0.1        0.01          1   \n",
       "7        DUX       1          1          1          1        0.01          1   \n",
       "8       ER71       1          1          1          1        0.01          1   \n",
       "9        ERG       1          1          1          1        0.01          1   \n",
       "10      ETS1       1          1          1          1        0.01          1   \n",
       "11      ETV1       1          1          1          1        0.01          1   \n",
       "12  EWS:FLI1       1          1          1          1        0.01          1   \n",
       "13       FLI       1          1          1          1        0.01          1   \n",
       "14      MYF5     0.1          1          1          1        0.01          1   \n",
       "15      MYOD     0.1          1          1        0.1        0.01          1   \n",
       "16      MYOG     0.1          1          1        0.1        0.01          1   \n",
       "17      PAX7       1          1          1          1        0.01          1   \n",
       "18     POLII       1          1          1          1        0.01          1   \n",
       "19     PPARG       1          1          1          1        0.01          1   \n",
       "20     RUNX2       1          1          1          1        0.01          1   \n",
       "21     TCF12    0.01          1          1        0.1        0.01          1   \n",
       "22    TWIST2     0.1          1          1        0.1        0.01          1   \n",
       "\n",
       "   Unnamed: 7 Unnamed: 8 Unnamed: 9  \n",
       "0      down1k   down1.5k     down2k  \n",
       "1           1          1          1  \n",
       "2           1          1          1  \n",
       "3         0.1          1          1  \n",
       "4           1          1          1  \n",
       "5           1          1          1  \n",
       "6           1          1          1  \n",
       "7           1          1          1  \n",
       "8           1          1          1  \n",
       "9           1          1          1  \n",
       "10          1          1          1  \n",
       "11          1          1          1  \n",
       "12          1          1          1  \n",
       "13          1          1          1  \n",
       "14        0.1          1          1  \n",
       "15          1          1          1  \n",
       "16          1          1        0.1  \n",
       "17          1          1          1  \n",
       "18          1          1          1  \n",
       "19          1          1          1  \n",
       "20          1          1          1  \n",
       "21        0.1          1          1  \n",
       "22          1          1          1  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_06 = pd.read_excel('pnas.2111920118.sd06.xlsx')\n",
    "data_06"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TSS-proximal DMR-coupled genes</th>\n",
       "      <th>TSS-dital-10k DMR-coupled genes</th>\n",
       "      <th>TSS-dital-100k DMR-coupled genes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>IL12RB1</td>\n",
       "      <td>GRAP2</td>\n",
       "      <td>TNIP3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>RUNX3</td>\n",
       "      <td>RUNX3</td>\n",
       "      <td>TUBD1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CARD11</td>\n",
       "      <td>ZBTB38</td>\n",
       "      <td>ERG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GRB2</td>\n",
       "      <td>GGA3</td>\n",
       "      <td>TMEM39A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ZPBP2</td>\n",
       "      <td>NPAS1</td>\n",
       "      <td>RUNX3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>FOXP1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCF7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>EPS15L1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PTPN13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>POU5F1B</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>76 rows  3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   TSS-proximal DMR-coupled genes TSS-dital-10k DMR-coupled genes  \\\n",
       "0                         IL12RB1                           GRAP2   \n",
       "1                           RUNX3                           RUNX3   \n",
       "2                          CARD11                          ZBTB38   \n",
       "3                            GRB2                            GGA3   \n",
       "4                           ZPBP2                           NPAS1   \n",
       "..                            ...                             ...   \n",
       "71                            NaN                             NaN   \n",
       "72                            NaN                             NaN   \n",
       "73                            NaN                             NaN   \n",
       "74                            NaN                             NaN   \n",
       "75                            NaN                             NaN   \n",
       "\n",
       "   TSS-dital-100k DMR-coupled genes  \n",
       "0                             TNIP3  \n",
       "1                             TUBD1  \n",
       "2                               ERG  \n",
       "3                           TMEM39A  \n",
       "4                             RUNX3  \n",
       "..                              ...  \n",
       "71                            FOXP1  \n",
       "72                             TCF7  \n",
       "73                          EPS15L1  \n",
       "74                           PTPN13  \n",
       "75                          POU5F1B  \n",
       "\n",
       "[76 rows x 3 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_07 = pd.read_excel('pnas.2111920118.sd07.xlsx')\n",
    "data_07"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene name</th>\n",
       "      <th>logFC</th>\n",
       "      <th>logCPM</th>\n",
       "      <th>P value</th>\n",
       "      <th>FDR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ARHGAP31</td>\n",
       "      <td>0.819578</td>\n",
       "      <td>4.471801</td>\n",
       "      <td>9.720000e-05</td>\n",
       "      <td>0.006011</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TRIB1</td>\n",
       "      <td>1.508544</td>\n",
       "      <td>6.125172</td>\n",
       "      <td>9.500000e-05</td>\n",
       "      <td>0.005918</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>E2F2</td>\n",
       "      <td>1.757939</td>\n",
       "      <td>3.132182</td>\n",
       "      <td>9.460000e-06</td>\n",
       "      <td>0.001635</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TAPBPL</td>\n",
       "      <td>0.625243</td>\n",
       "      <td>4.549739</td>\n",
       "      <td>9.110000e-05</td>\n",
       "      <td>0.005695</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MYO1D</td>\n",
       "      <td>1.491649</td>\n",
       "      <td>4.945613</td>\n",
       "      <td>9.100000e-07</td>\n",
       "      <td>0.000398</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>542</th>\n",
       "      <td>APBB1</td>\n",
       "      <td>-0.695130</td>\n",
       "      <td>4.786609</td>\n",
       "      <td>1.269600e-04</td>\n",
       "      <td>0.007116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>543</th>\n",
       "      <td>H1-3</td>\n",
       "      <td>-2.155091</td>\n",
       "      <td>1.245875</td>\n",
       "      <td>1.258320e-04</td>\n",
       "      <td>0.007116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>544</th>\n",
       "      <td>BEX4</td>\n",
       "      <td>-0.709985</td>\n",
       "      <td>4.886337</td>\n",
       "      <td>1.213630e-04</td>\n",
       "      <td>0.007106</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>545</th>\n",
       "      <td>SECISBP2L</td>\n",
       "      <td>-0.581726</td>\n",
       "      <td>7.028738</td>\n",
       "      <td>1.190150e-04</td>\n",
       "      <td>0.006997</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>546</th>\n",
       "      <td>RCN2</td>\n",
       "      <td>-0.687891</td>\n",
       "      <td>5.345495</td>\n",
       "      <td>1.018960e-04</td>\n",
       "      <td>0.006207</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>547 rows  5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Gene name     logFC    logCPM       P value       FDR\n",
       "0     ARHGAP31  0.819578  4.471801  9.720000e-05  0.006011\n",
       "1        TRIB1  1.508544  6.125172  9.500000e-05  0.005918\n",
       "2         E2F2  1.757939  3.132182  9.460000e-06  0.001635\n",
       "3       TAPBPL  0.625243  4.549739  9.110000e-05  0.005695\n",
       "4        MYO1D  1.491649  4.945613  9.100000e-07  0.000398\n",
       "..         ...       ...       ...           ...       ...\n",
       "542      APBB1 -0.695130  4.786609  1.269600e-04  0.007116\n",
       "543       H1-3 -2.155091  1.245875  1.258320e-04  0.007116\n",
       "544       BEX4 -0.709985  4.886337  1.213630e-04  0.007106\n",
       "545  SECISBP2L -0.581726  7.028738  1.190150e-04  0.006997\n",
       "546       RCN2 -0.687891  5.345495  1.018960e-04  0.006207\n",
       "\n",
       "[547 rows x 5 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read data\n",
    "data_08 = pd.read_excel('pnas.2111920118.sd08.xlsx')\n",
    "data_08"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GroupID</th>\n",
       "      <th>Category</th>\n",
       "      <th>Term</th>\n",
       "      <th>Description</th>\n",
       "      <th>LogP</th>\n",
       "      <th>Log(q-value)</th>\n",
       "      <th>InTerm_InList</th>\n",
       "      <th>Genes</th>\n",
       "      <th>Symbols</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1_Summary</td>\n",
       "      <td>GO Biological Processes</td>\n",
       "      <td>GO:0006958</td>\n",
       "      <td>complement activation, classical pathway</td>\n",
       "      <td>-48.304348</td>\n",
       "      <td>-44.107481</td>\n",
       "      <td>46/141</td>\n",
       "      <td>1191,3493,3494,3500,3501,3503,3514,3538,3539,2...</td>\n",
       "      <td>CLU,IGHA1,IGHA2,IGHG1,IGHG2,IGHG4,IGKC,IGLC2,I...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1_Member</td>\n",
       "      <td>GO Biological Processes</td>\n",
       "      <td>GO:0006958</td>\n",
       "      <td>complement activation, classical pathway</td>\n",
       "      <td>-48.304348</td>\n",
       "      <td>-44.107481</td>\n",
       "      <td>46/141</td>\n",
       "      <td>1191,3493,3494,3500,3501,3503,3514,3538,3539,2...</td>\n",
       "      <td>CLU,IGHA1,IGHA2,IGHG1,IGHG2,IGHG4,IGKC,IGLC2,I...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1_Member</td>\n",
       "      <td>GO Biological Processes</td>\n",
       "      <td>GO:0002455</td>\n",
       "      <td>humoral immune response mediated by circulatin...</td>\n",
       "      <td>-47.484893</td>\n",
       "      <td>-43.589056</td>\n",
       "      <td>47/156</td>\n",
       "      <td>1191,3493,3494,3500,3501,3503,3514,3538,3539,9...</td>\n",
       "      <td>CLU,IGHA1,IGHA2,IGHG1,IGHG2,IGHG4,IGKC,IGLC2,I...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1_Member</td>\n",
       "      <td>GO Biological Processes</td>\n",
       "      <td>GO:0016064</td>\n",
       "      <td>immunoglobulin mediated immune response</td>\n",
       "      <td>-43.441447</td>\n",
       "      <td>-39.721701</td>\n",
       "      <td>50/223</td>\n",
       "      <td>939,1191,3493,3494,3500,3501,3503,3514,3538,35...</td>\n",
       "      <td>CD27,CLU,IGHA1,IGHA2,IGHG1,IGHG2,IGHG4,IGKC,IG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1_Member</td>\n",
       "      <td>GO Biological Processes</td>\n",
       "      <td>GO:0019724</td>\n",
       "      <td>B cell mediated immunity</td>\n",
       "      <td>-43.130850</td>\n",
       "      <td>-39.536043</td>\n",
       "      <td>50/226</td>\n",
       "      <td>939,1191,3493,3494,3500,3501,3503,3514,3538,35...</td>\n",
       "      <td>CD27,CLU,IGHA1,IGHA2,IGHG1,IGHG2,IGHG4,IGKC,IG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>292</th>\n",
       "      <td>19_Summary</td>\n",
       "      <td>GO Biological Processes</td>\n",
       "      <td>GO:0007219</td>\n",
       "      <td>Notch signaling pathway</td>\n",
       "      <td>-3.959462</td>\n",
       "      <td>-1.932857</td>\n",
       "      <td>11/186</td>\n",
       "      <td>3559,5087,5584,6772,7471,8470,8626,9787,64065,...</td>\n",
       "      <td>IL2RA,PBX1,PRKCI,STAT1,WNT1,SORBS2,TP63,DLGAP5...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>293</th>\n",
       "      <td>19_Member</td>\n",
       "      <td>GO Biological Processes</td>\n",
       "      <td>GO:0007219</td>\n",
       "      <td>Notch signaling pathway</td>\n",
       "      <td>-3.959462</td>\n",
       "      <td>-1.932857</td>\n",
       "      <td>11/186</td>\n",
       "      <td>3559,5087,5584,6772,7471,8470,8626,9787,64065,...</td>\n",
       "      <td>IL2RA,PBX1,PRKCI,STAT1,WNT1,SORBS2,TP63,DLGAP5...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294</th>\n",
       "      <td>19_Member</td>\n",
       "      <td>GO Biological Processes</td>\n",
       "      <td>GO:0007221</td>\n",
       "      <td>positive regulation of transcription of Notch ...</td>\n",
       "      <td>-2.653534</td>\n",
       "      <td>-0.908453</td>\n",
       "      <td>3/18</td>\n",
       "      <td>5087,6772,9787</td>\n",
       "      <td>PBX1,STAT1,DLGAP5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>295</th>\n",
       "      <td>20_Summary</td>\n",
       "      <td>GO Biological Processes</td>\n",
       "      <td>GO:1904587</td>\n",
       "      <td>response to glycoprotein</td>\n",
       "      <td>-3.769281</td>\n",
       "      <td>-1.765539</td>\n",
       "      <td>3/8</td>\n",
       "      <td>4121,7253,80267</td>\n",
       "      <td>MAN1A1,TSHR,EDEM3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>296</th>\n",
       "      <td>20_Member</td>\n",
       "      <td>GO Biological Processes</td>\n",
       "      <td>GO:1904587</td>\n",
       "      <td>response to glycoprotein</td>\n",
       "      <td>-3.769281</td>\n",
       "      <td>-1.765539</td>\n",
       "      <td>3/8</td>\n",
       "      <td>4121,7253,80267</td>\n",
       "      <td>MAN1A1,TSHR,EDEM3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>297 rows  9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        GroupID                 Category        Term  \\\n",
       "0     1_Summary  GO Biological Processes  GO:0006958   \n",
       "1      1_Member  GO Biological Processes  GO:0006958   \n",
       "2      1_Member  GO Biological Processes  GO:0002455   \n",
       "3      1_Member  GO Biological Processes  GO:0016064   \n",
       "4      1_Member  GO Biological Processes  GO:0019724   \n",
       "..          ...                      ...         ...   \n",
       "292  19_Summary  GO Biological Processes  GO:0007219   \n",
       "293   19_Member  GO Biological Processes  GO:0007219   \n",
       "294   19_Member  GO Biological Processes  GO:0007221   \n",
       "295  20_Summary  GO Biological Processes  GO:1904587   \n",
       "296   20_Member  GO Biological Processes  GO:1904587   \n",
       "\n",
       "                                           Description       LogP  \\\n",
       "0             complement activation, classical pathway -48.304348   \n",
       "1             complement activation, classical pathway -48.304348   \n",
       "2    humoral immune response mediated by circulatin... -47.484893   \n",
       "3              immunoglobulin mediated immune response -43.441447   \n",
       "4                             B cell mediated immunity -43.130850   \n",
       "..                                                 ...        ...   \n",
       "292                            Notch signaling pathway  -3.959462   \n",
       "293                            Notch signaling pathway  -3.959462   \n",
       "294  positive regulation of transcription of Notch ...  -2.653534   \n",
       "295                           response to glycoprotein  -3.769281   \n",
       "296                           response to glycoprotein  -3.769281   \n",
       "\n",
       "     Log(q-value) InTerm_InList  \\\n",
       "0      -44.107481        46/141   \n",
       "1      -44.107481        46/141   \n",
       "2      -43.589056        47/156   \n",
       "3      -39.721701        50/223   \n",
       "4      -39.536043        50/226   \n",
       "..            ...           ...   \n",
       "292     -1.932857        11/186   \n",
       "293     -1.932857        11/186   \n",
       "294     -0.908453          3/18   \n",
       "295     -1.765539           3/8   \n",
       "296     -1.765539           3/8   \n",
       "\n",
       "                                                 Genes  \\\n",
       "0    1191,3493,3494,3500,3501,3503,3514,3538,3539,2...   \n",
       "1    1191,3493,3494,3500,3501,3503,3514,3538,3539,2...   \n",
       "2    1191,3493,3494,3500,3501,3503,3514,3538,3539,9...   \n",
       "3    939,1191,3493,3494,3500,3501,3503,3514,3538,35...   \n",
       "4    939,1191,3493,3494,3500,3501,3503,3514,3538,35...   \n",
       "..                                                 ...   \n",
       "292  3559,5087,5584,6772,7471,8470,8626,9787,64065,...   \n",
       "293  3559,5087,5584,6772,7471,8470,8626,9787,64065,...   \n",
       "294                                     5087,6772,9787   \n",
       "295                                    4121,7253,80267   \n",
       "296                                    4121,7253,80267   \n",
       "\n",
       "                                               Symbols  \n",
       "0    CLU,IGHA1,IGHA2,IGHG1,IGHG2,IGHG4,IGKC,IGLC2,I...  \n",
       "1    CLU,IGHA1,IGHA2,IGHG1,IGHG2,IGHG4,IGKC,IGLC2,I...  \n",
       "2    CLU,IGHA1,IGHA2,IGHG1,IGHG2,IGHG4,IGKC,IGLC2,I...  \n",
       "3    CD27,CLU,IGHA1,IGHA2,IGHG1,IGHG2,IGHG4,IGKC,IG...  \n",
       "4    CD27,CLU,IGHA1,IGHA2,IGHG1,IGHG2,IGHG4,IGKC,IG...  \n",
       "..                                                 ...  \n",
       "292  IL2RA,PBX1,PRKCI,STAT1,WNT1,SORBS2,TP63,DLGAP5...  \n",
       "293  IL2RA,PBX1,PRKCI,STAT1,WNT1,SORBS2,TP63,DLGAP5...  \n",
       "294                                  PBX1,STAT1,DLGAP5  \n",
       "295                                  MAN1A1,TSHR,EDEM3  \n",
       "296                                  MAN1A1,TSHR,EDEM3  \n",
       "\n",
       "[297 rows x 9 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_09 = pd.read_excel('pnas.2111920118.sd09.xlsx')\n",
    "data_09"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GroupID</th>\n",
       "      <th>Category</th>\n",
       "      <th>Term</th>\n",
       "      <th>Description</th>\n",
       "      <th>LogP</th>\n",
       "      <th>Log(q-value)</th>\n",
       "      <th>InTerm_InList</th>\n",
       "      <th>Genes</th>\n",
       "      <th>Symbols</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1_Summary</td>\n",
       "      <td>GO Biological Processes</td>\n",
       "      <td>GO:0006334</td>\n",
       "      <td>nucleosome assembly</td>\n",
       "      <td>-6.432346</td>\n",
       "      <td>-2.246902</td>\n",
       "      <td>8/145</td>\n",
       "      <td>3006,3007,3008,3017,8339,8349,8971,440689,3012...</td>\n",
       "      <td>H1-2,H1-3,H1-4,H2BC5,H2BC8,H2BC21,H1-10,H2BC18...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1_Member</td>\n",
       "      <td>GO Biological Processes</td>\n",
       "      <td>GO:0006334</td>\n",
       "      <td>nucleosome assembly</td>\n",
       "      <td>-6.432346</td>\n",
       "      <td>-2.246902</td>\n",
       "      <td>8/145</td>\n",
       "      <td>3006,3007,3008,3017,8339,8349,8971,440689</td>\n",
       "      <td>H1-2,H1-3,H1-4,H2BC5,H2BC8,H2BC21,H1-10,H2BC18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1_Member</td>\n",
       "      <td>GO Biological Processes</td>\n",
       "      <td>GO:0016584</td>\n",
       "      <td>nucleosome positioning</td>\n",
       "      <td>-6.133274</td>\n",
       "      <td>-2.246902</td>\n",
       "      <td>4/16</td>\n",
       "      <td>3006,3007,3008,8971</td>\n",
       "      <td>H1-2,H1-3,H1-4,H1-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1_Member</td>\n",
       "      <td>GO Biological Processes</td>\n",
       "      <td>GO:0031497</td>\n",
       "      <td>chromatin assembly</td>\n",
       "      <td>-5.966648</td>\n",
       "      <td>-2.246902</td>\n",
       "      <td>8/167</td>\n",
       "      <td>3006,3007,3008,3017,8339,8349,8971,440689</td>\n",
       "      <td>H1-2,H1-3,H1-4,H2BC5,H2BC8,H2BC21,H1-10,H2BC18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1_Member</td>\n",
       "      <td>GO Biological Processes</td>\n",
       "      <td>GO:0031936</td>\n",
       "      <td>negative regulation of chromatin silencing</td>\n",
       "      <td>-5.809595</td>\n",
       "      <td>-2.214788</td>\n",
       "      <td>4/19</td>\n",
       "      <td>3006,3007,3008,8971</td>\n",
       "      <td>H1-2,H1-3,H1-4,H1-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135</th>\n",
       "      <td>19_Member</td>\n",
       "      <td>GO Biological Processes</td>\n",
       "      <td>GO:0044089</td>\n",
       "      <td>positive regulation of cellular component biog...</td>\n",
       "      <td>-2.373029</td>\n",
       "      <td>-0.037347</td>\n",
       "      <td>8/557</td>\n",
       "      <td>19,5290,8027,8739,9026,54413,54922,123720</td>\n",
       "      <td>ABCA1,PIK3CA,STAM,HRK,HIP1R,NLGN3,RASIP1,WHAMM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>19_Member</td>\n",
       "      <td>GO Biological Processes</td>\n",
       "      <td>GO:0043254</td>\n",
       "      <td>regulation of protein complex assembly</td>\n",
       "      <td>-2.112964</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>7/492</td>\n",
       "      <td>19,5494,8739,9026,54922,90952,123720</td>\n",
       "      <td>ABCA1,PPM1A,HRK,HIP1R,RASIP1,ESAM,WHAMM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137</th>\n",
       "      <td>20_Summary</td>\n",
       "      <td>GO Biological Processes</td>\n",
       "      <td>GO:0007050</td>\n",
       "      <td>cell cycle arrest</td>\n",
       "      <td>-2.325913</td>\n",
       "      <td>-0.021141</td>\n",
       "      <td>5/236</td>\n",
       "      <td>322,323,5494,9337,123720,3087,7159,10950</td>\n",
       "      <td>APBB1,APBB2,PPM1A,CNOT8,WHAMM,HHEX,TP53BP2,BTG3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138</th>\n",
       "      <td>20_Member</td>\n",
       "      <td>GO Biological Processes</td>\n",
       "      <td>GO:0007050</td>\n",
       "      <td>cell cycle arrest</td>\n",
       "      <td>-2.325913</td>\n",
       "      <td>-0.021141</td>\n",
       "      <td>5/236</td>\n",
       "      <td>322,323,5494,9337,123720</td>\n",
       "      <td>APBB1,APBB2,PPM1A,CNOT8,WHAMM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>20_Member</td>\n",
       "      <td>GO Biological Processes</td>\n",
       "      <td>GO:0045786</td>\n",
       "      <td>negative regulation of cell cycle</td>\n",
       "      <td>-2.073825</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>8/627</td>\n",
       "      <td>322,323,3087,5494,7159,9337,10950,123720</td>\n",
       "      <td>APBB1,APBB2,HHEX,PPM1A,TP53BP2,CNOT8,BTG3,WHAMM</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>140 rows  9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        GroupID                 Category        Term  \\\n",
       "0     1_Summary  GO Biological Processes  GO:0006334   \n",
       "1      1_Member  GO Biological Processes  GO:0006334   \n",
       "2      1_Member  GO Biological Processes  GO:0016584   \n",
       "3      1_Member  GO Biological Processes  GO:0031497   \n",
       "4      1_Member  GO Biological Processes  GO:0031936   \n",
       "..          ...                      ...         ...   \n",
       "135   19_Member  GO Biological Processes  GO:0044089   \n",
       "136   19_Member  GO Biological Processes  GO:0043254   \n",
       "137  20_Summary  GO Biological Processes  GO:0007050   \n",
       "138   20_Member  GO Biological Processes  GO:0007050   \n",
       "139   20_Member  GO Biological Processes  GO:0045786   \n",
       "\n",
       "                                           Description      LogP  \\\n",
       "0                                  nucleosome assembly -6.432346   \n",
       "1                                  nucleosome assembly -6.432346   \n",
       "2                               nucleosome positioning -6.133274   \n",
       "3                                   chromatin assembly -5.966648   \n",
       "4           negative regulation of chromatin silencing -5.809595   \n",
       "..                                                 ...       ...   \n",
       "135  positive regulation of cellular component biog... -2.373029   \n",
       "136             regulation of protein complex assembly -2.112964   \n",
       "137                                  cell cycle arrest -2.325913   \n",
       "138                                  cell cycle arrest -2.325913   \n",
       "139                  negative regulation of cell cycle -2.073825   \n",
       "\n",
       "     Log(q-value) InTerm_InList  \\\n",
       "0       -2.246902         8/145   \n",
       "1       -2.246902         8/145   \n",
       "2       -2.246902          4/16   \n",
       "3       -2.246902         8/167   \n",
       "4       -2.214788          4/19   \n",
       "..            ...           ...   \n",
       "135     -0.037347         8/557   \n",
       "136      0.000000         7/492   \n",
       "137     -0.021141         5/236   \n",
       "138     -0.021141         5/236   \n",
       "139      0.000000         8/627   \n",
       "\n",
       "                                                 Genes  \\\n",
       "0    3006,3007,3008,3017,8339,8349,8971,440689,3012...   \n",
       "1            3006,3007,3008,3017,8339,8349,8971,440689   \n",
       "2                                  3006,3007,3008,8971   \n",
       "3            3006,3007,3008,3017,8339,8349,8971,440689   \n",
       "4                                  3006,3007,3008,8971   \n",
       "..                                                 ...   \n",
       "135          19,5290,8027,8739,9026,54413,54922,123720   \n",
       "136               19,5494,8739,9026,54922,90952,123720   \n",
       "137           322,323,5494,9337,123720,3087,7159,10950   \n",
       "138                           322,323,5494,9337,123720   \n",
       "139           322,323,3087,5494,7159,9337,10950,123720   \n",
       "\n",
       "                                               Symbols  \n",
       "0    H1-2,H1-3,H1-4,H2BC5,H2BC8,H2BC21,H1-10,H2BC18...  \n",
       "1       H1-2,H1-3,H1-4,H2BC5,H2BC8,H2BC21,H1-10,H2BC18  \n",
       "2                                 H1-2,H1-3,H1-4,H1-10  \n",
       "3       H1-2,H1-3,H1-4,H2BC5,H2BC8,H2BC21,H1-10,H2BC18  \n",
       "4                                 H1-2,H1-3,H1-4,H1-10  \n",
       "..                                                 ...  \n",
       "135     ABCA1,PIK3CA,STAM,HRK,HIP1R,NLGN3,RASIP1,WHAMM  \n",
       "136            ABCA1,PPM1A,HRK,HIP1R,RASIP1,ESAM,WHAMM  \n",
       "137    APBB1,APBB2,PPM1A,CNOT8,WHAMM,HHEX,TP53BP2,BTG3  \n",
       "138                      APBB1,APBB2,PPM1A,CNOT8,WHAMM  \n",
       "139    APBB1,APBB2,HHEX,PPM1A,TP53BP2,CNOT8,BTG3,WHAMM  \n",
       "\n",
       "[140 rows x 9 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_10 = pd.read_excel('pnas.2111920118.sd10.xlsx')\n",
    "data_10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene_name</th>\n",
       "      <th>GeneDesc</th>\n",
       "      <th>Function</th>\n",
       "      <th>Modification</th>\n",
       "      <th>Target</th>\n",
       "      <th>Specific_target</th>\n",
       "      <th>Product</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>HJURP</td>\n",
       "      <td>#</td>\n",
       "      <td>Histone chaperone</td>\n",
       "      <td>#</td>\n",
       "      <td>histone</td>\n",
       "      <td>H4</td>\n",
       "      <td>#</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>RNF168</td>\n",
       "      <td>RING-type (C3HC4) zinc fingers</td>\n",
       "      <td>Histone modification write</td>\n",
       "      <td>Histone ubiquitination</td>\n",
       "      <td>histone</td>\n",
       "      <td>H2AK13, H2AK15, H2AXK13, H2AXK15</td>\n",
       "      <td>H2AK13ub, H2AK15ub, H2AXK13ub, H2AXK15ub</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TTK</td>\n",
       "      <td>#</td>\n",
       "      <td>Histone modification write cofactor</td>\n",
       "      <td>Histone phosphorylation</td>\n",
       "      <td>histone</td>\n",
       "      <td>H2AT120</td>\n",
       "      <td>H2AT120ph</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PRDM1</td>\n",
       "      <td>Zinc fingers, C2H2-type</td>\n",
       "      <td>Histone modification write cofactor</td>\n",
       "      <td>Histone methylation</td>\n",
       "      <td>histone</td>\n",
       "      <td>H3K9</td>\n",
       "      <td>H3K9me</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GFI1B</td>\n",
       "      <td>Zinc fingers, C2H2-type</td>\n",
       "      <td>Histone modification cofactor</td>\n",
       "      <td>#</td>\n",
       "      <td>#</td>\n",
       "      <td>#</td>\n",
       "      <td>#</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>VDR</td>\n",
       "      <td>Nuclear hormone receptors</td>\n",
       "      <td>Chromatin remodeling cofactor, TF</td>\n",
       "      <td>#</td>\n",
       "      <td>histone</td>\n",
       "      <td>H2BK12ac, H3K14ac, H4K16ac</td>\n",
       "      <td>#</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>CIT</td>\n",
       "      <td>#</td>\n",
       "      <td>Histone modification write cofactor, Histone m...</td>\n",
       "      <td>Histone methylation, Histone phosphorylation</td>\n",
       "      <td>histone</td>\n",
       "      <td>H3K9</td>\n",
       "      <td>H3K9me</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>AURKB</td>\n",
       "      <td>Serine/threonine phosphatases / Protein phosph...</td>\n",
       "      <td>Histone modification write</td>\n",
       "      <td>Histone phosphorylation</td>\n",
       "      <td>histone</td>\n",
       "      <td>H3S10, H3S28</td>\n",
       "      <td>#</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>CDC6</td>\n",
       "      <td>#</td>\n",
       "      <td>Chromatin remodeling</td>\n",
       "      <td>#</td>\n",
       "      <td>chromatin</td>\n",
       "      <td>#</td>\n",
       "      <td>#</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>TOP2A</td>\n",
       "      <td>#</td>\n",
       "      <td>Chromatin remodeling</td>\n",
       "      <td>#</td>\n",
       "      <td>DNA</td>\n",
       "      <td>DNA</td>\n",
       "      <td>#</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>KEAP1</td>\n",
       "      <td>Kelch-like, \"BTB/POZ domain containing\"</td>\n",
       "      <td>Chromatin remodeling</td>\n",
       "      <td>#</td>\n",
       "      <td>chromatin</td>\n",
       "      <td>#</td>\n",
       "      <td>#</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ASF1B</td>\n",
       "      <td>#</td>\n",
       "      <td>Histone chaperone</td>\n",
       "      <td>#</td>\n",
       "      <td>histone</td>\n",
       "      <td>H3, H4</td>\n",
       "      <td>#</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>APOBEC3B</td>\n",
       "      <td>Apolipoprotein B mRNA editing enzymes</td>\n",
       "      <td>DNA modification, RNA modification</td>\n",
       "      <td>DNA demethylation, mRNA editing</td>\n",
       "      <td>DNA, RNA</td>\n",
       "      <td>ssDNA, mRNA, mC</td>\n",
       "      <td>dhU</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>KDM5B</td>\n",
       "      <td>Chromatin-modifying enzymes / K-demethylases, ...</td>\n",
       "      <td>Histone modification erase</td>\n",
       "      <td>Histone methylation</td>\n",
       "      <td>histone</td>\n",
       "      <td>H3K4me3</td>\n",
       "      <td>H3K4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>SETD7</td>\n",
       "      <td>Chromatin-modifying enzymes / K-methyltransfer...</td>\n",
       "      <td>Histone modification write</td>\n",
       "      <td>Histone methylation</td>\n",
       "      <td>histone</td>\n",
       "      <td>H3K4</td>\n",
       "      <td>H3K4me1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>APBB1</td>\n",
       "      <td>#</td>\n",
       "      <td>Histone modification</td>\n",
       "      <td>#</td>\n",
       "      <td>histone</td>\n",
       "      <td>H2AX</td>\n",
       "      <td>#</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>FOXO1</td>\n",
       "      <td>Forkhead boxes</td>\n",
       "      <td>TF</td>\n",
       "      <td>#</td>\n",
       "      <td>histone, DNA</td>\n",
       "      <td>DNA motif, H3, H4</td>\n",
       "      <td>#</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Gene_name                                           GeneDesc  \\\n",
       "0      HJURP                                                  #   \n",
       "1     RNF168                     RING-type (C3HC4) zinc fingers   \n",
       "2        TTK                                                  #   \n",
       "3      PRDM1                            Zinc fingers, C2H2-type   \n",
       "4      GFI1B                            Zinc fingers, C2H2-type   \n",
       "5        VDR                          Nuclear hormone receptors   \n",
       "6        CIT                                                  #   \n",
       "7      AURKB  Serine/threonine phosphatases / Protein phosph...   \n",
       "8       CDC6                                                  #   \n",
       "9      TOP2A                                                  #   \n",
       "10     KEAP1            Kelch-like, \"BTB/POZ domain containing\"   \n",
       "11     ASF1B                                                  #   \n",
       "12  APOBEC3B              Apolipoprotein B mRNA editing enzymes   \n",
       "13     KDM5B  Chromatin-modifying enzymes / K-demethylases, ...   \n",
       "14     SETD7  Chromatin-modifying enzymes / K-methyltransfer...   \n",
       "15     APBB1                                                  #   \n",
       "16     FOXO1                                     Forkhead boxes   \n",
       "\n",
       "                                             Function  \\\n",
       "0                                   Histone chaperone   \n",
       "1                          Histone modification write   \n",
       "2                 Histone modification write cofactor   \n",
       "3                 Histone modification write cofactor   \n",
       "4                       Histone modification cofactor   \n",
       "5                   Chromatin remodeling cofactor, TF   \n",
       "6   Histone modification write cofactor, Histone m...   \n",
       "7                          Histone modification write   \n",
       "8                                Chromatin remodeling   \n",
       "9                                Chromatin remodeling   \n",
       "10                               Chromatin remodeling   \n",
       "11                                  Histone chaperone   \n",
       "12                 DNA modification, RNA modification   \n",
       "13                         Histone modification erase   \n",
       "14                         Histone modification write   \n",
       "15                               Histone modification   \n",
       "16                                                 TF   \n",
       "\n",
       "                                    Modification        Target  \\\n",
       "0                                              #       histone   \n",
       "1                         Histone ubiquitination       histone   \n",
       "2                        Histone phosphorylation       histone   \n",
       "3                            Histone methylation       histone   \n",
       "4                                              #             #   \n",
       "5                                              #       histone   \n",
       "6   Histone methylation, Histone phosphorylation       histone   \n",
       "7                        Histone phosphorylation       histone   \n",
       "8                                              #     chromatin   \n",
       "9                                              #           DNA   \n",
       "10                                             #     chromatin   \n",
       "11                                             #       histone   \n",
       "12               DNA demethylation, mRNA editing      DNA, RNA   \n",
       "13                           Histone methylation       histone   \n",
       "14                           Histone methylation       histone   \n",
       "15                                             #       histone   \n",
       "16                                             #  histone, DNA   \n",
       "\n",
       "                     Specific_target                                   Product  \n",
       "0                                 H4                                         #  \n",
       "1   H2AK13, H2AK15, H2AXK13, H2AXK15  H2AK13ub, H2AK15ub, H2AXK13ub, H2AXK15ub  \n",
       "2                            H2AT120                                 H2AT120ph  \n",
       "3                               H3K9                                    H3K9me  \n",
       "4                                  #                                         #  \n",
       "5         H2BK12ac, H3K14ac, H4K16ac                                         #  \n",
       "6                               H3K9                                    H3K9me  \n",
       "7                       H3S10, H3S28                                         #  \n",
       "8                                  #                                         #  \n",
       "9                                DNA                                         #  \n",
       "10                                 #                                         #  \n",
       "11                            H3, H4                                         #  \n",
       "12                   ssDNA, mRNA, mC                                       dhU  \n",
       "13                           H3K4me3                                      H3K4  \n",
       "14                              H3K4                                   H3K4me1  \n",
       "15                              H2AX                                         #  \n",
       "16                 DNA motif, H3, H4                                         #  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_11 = pd.read_excel('pnas.2111920118.sd11.xlsx')\n",
    "data_11"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GroupID</th>\n",
       "      <th>Category</th>\n",
       "      <th>Term</th>\n",
       "      <th>Description</th>\n",
       "      <th>LogP</th>\n",
       "      <th>Log(q-value)</th>\n",
       "      <th>InTerm_InList</th>\n",
       "      <th>Genes</th>\n",
       "      <th>Symbols</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1_Summary</td>\n",
       "      <td>GO Biological Processes</td>\n",
       "      <td>GO:0046649</td>\n",
       "      <td>lymphocyte activation</td>\n",
       "      <td>-4.317160</td>\n",
       "      <td>-0.543740</td>\n",
       "      <td>11/754</td>\n",
       "      <td>639,939,2175,3688,6693,8324,23495,23705,51237,...</td>\n",
       "      <td>PRDM1,CD27,FANCA,ITGB1,SPN,FZD7,TNFRSF13B,CADM...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1_Member</td>\n",
       "      <td>GO Biological Processes</td>\n",
       "      <td>GO:0046649</td>\n",
       "      <td>lymphocyte activation</td>\n",
       "      <td>-4.317160</td>\n",
       "      <td>-0.543740</td>\n",
       "      <td>11/754</td>\n",
       "      <td>639,939,2175,3688,6693,8324,23495,23705,51237,...</td>\n",
       "      <td>PRDM1,CD27,FANCA,ITGB1,SPN,FZD7,TNFRSF13B,CADM...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1_Member</td>\n",
       "      <td>GO Biological Processes</td>\n",
       "      <td>GO:0030217</td>\n",
       "      <td>T cell differentiation</td>\n",
       "      <td>-3.719172</td>\n",
       "      <td>-0.543740</td>\n",
       "      <td>6/246</td>\n",
       "      <td>639,939,2175,6693,8324,64218</td>\n",
       "      <td>PRDM1,CD27,FANCA,SPN,FZD7,SEMA4A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1_Member</td>\n",
       "      <td>GO Biological Processes</td>\n",
       "      <td>GO:0030098</td>\n",
       "      <td>lymphocyte differentiation</td>\n",
       "      <td>-3.666057</td>\n",
       "      <td>-0.543740</td>\n",
       "      <td>7/358</td>\n",
       "      <td>639,939,2175,3688,6693,8324,64218</td>\n",
       "      <td>PRDM1,CD27,FANCA,ITGB1,SPN,FZD7,SEMA4A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1_Member</td>\n",
       "      <td>GO Biological Processes</td>\n",
       "      <td>GO:0002521</td>\n",
       "      <td>leukocyte differentiation</td>\n",
       "      <td>-3.382330</td>\n",
       "      <td>-0.539877</td>\n",
       "      <td>8/526</td>\n",
       "      <td>639,939,2175,3688,6693,8324,10221,64218</td>\n",
       "      <td>PRDM1,CD27,FANCA,ITGB1,SPN,FZD7,TRIB1,SEMA4A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164</th>\n",
       "      <td>19_Member</td>\n",
       "      <td>GO Biological Processes</td>\n",
       "      <td>GO:0030278</td>\n",
       "      <td>regulation of ossification</td>\n",
       "      <td>-2.254557</td>\n",
       "      <td>-0.041364</td>\n",
       "      <td>4/209</td>\n",
       "      <td>860,5087,5552,6096</td>\n",
       "      <td>RUNX2,PBX1,SRGN,RORB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165</th>\n",
       "      <td>20_Summary</td>\n",
       "      <td>GO Biological Processes</td>\n",
       "      <td>GO:0016485</td>\n",
       "      <td>protein processing</td>\n",
       "      <td>-2.234065</td>\n",
       "      <td>-0.041364</td>\n",
       "      <td>5/338</td>\n",
       "      <td>1191,2903,4241,5552,90701,1509,10221,10257</td>\n",
       "      <td>CLU,GRIN2A,MELTF,SRGN,SEC11C,CTSD,TRIB1,ABCC4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>166</th>\n",
       "      <td>20_Member</td>\n",
       "      <td>GO Biological Processes</td>\n",
       "      <td>GO:0016485</td>\n",
       "      <td>protein processing</td>\n",
       "      <td>-2.234065</td>\n",
       "      <td>-0.041364</td>\n",
       "      <td>5/338</td>\n",
       "      <td>1191,2903,4241,5552,90701</td>\n",
       "      <td>CLU,GRIN2A,MELTF,SRGN,SEC11C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>167</th>\n",
       "      <td>20_Member</td>\n",
       "      <td>GO Biological Processes</td>\n",
       "      <td>GO:0045862</td>\n",
       "      <td>positive regulation of proteolysis</td>\n",
       "      <td>-2.035290</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>5/378</td>\n",
       "      <td>1191,1509,2903,4241,10221</td>\n",
       "      <td>CLU,CTSD,GRIN2A,MELTF,TRIB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>168</th>\n",
       "      <td>20_Member</td>\n",
       "      <td>GO Biological Processes</td>\n",
       "      <td>GO:0002576</td>\n",
       "      <td>platelet degranulation</td>\n",
       "      <td>-2.027199</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3/128</td>\n",
       "      <td>1191,5552,10257</td>\n",
       "      <td>CLU,SRGN,ABCC4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>169 rows  9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        GroupID                 Category        Term  \\\n",
       "0     1_Summary  GO Biological Processes  GO:0046649   \n",
       "1      1_Member  GO Biological Processes  GO:0046649   \n",
       "2      1_Member  GO Biological Processes  GO:0030217   \n",
       "3      1_Member  GO Biological Processes  GO:0030098   \n",
       "4      1_Member  GO Biological Processes  GO:0002521   \n",
       "..          ...                      ...         ...   \n",
       "164   19_Member  GO Biological Processes  GO:0030278   \n",
       "165  20_Summary  GO Biological Processes  GO:0016485   \n",
       "166   20_Member  GO Biological Processes  GO:0016485   \n",
       "167   20_Member  GO Biological Processes  GO:0045862   \n",
       "168   20_Member  GO Biological Processes  GO:0002576   \n",
       "\n",
       "                            Description      LogP  Log(q-value) InTerm_InList  \\\n",
       "0                 lymphocyte activation -4.317160     -0.543740        11/754   \n",
       "1                 lymphocyte activation -4.317160     -0.543740        11/754   \n",
       "2                T cell differentiation -3.719172     -0.543740         6/246   \n",
       "3            lymphocyte differentiation -3.666057     -0.543740         7/358   \n",
       "4             leukocyte differentiation -3.382330     -0.539877         8/526   \n",
       "..                                  ...       ...           ...           ...   \n",
       "164          regulation of ossification -2.254557     -0.041364         4/209   \n",
       "165                  protein processing -2.234065     -0.041364         5/338   \n",
       "166                  protein processing -2.234065     -0.041364         5/338   \n",
       "167  positive regulation of proteolysis -2.035290      0.000000         5/378   \n",
       "168              platelet degranulation -2.027199      0.000000         3/128   \n",
       "\n",
       "                                                 Genes  \\\n",
       "0    639,939,2175,3688,6693,8324,23495,23705,51237,...   \n",
       "1    639,939,2175,3688,6693,8324,23495,23705,51237,...   \n",
       "2                         639,939,2175,6693,8324,64218   \n",
       "3                    639,939,2175,3688,6693,8324,64218   \n",
       "4              639,939,2175,3688,6693,8324,10221,64218   \n",
       "..                                                 ...   \n",
       "164                                 860,5087,5552,6096   \n",
       "165         1191,2903,4241,5552,90701,1509,10221,10257   \n",
       "166                          1191,2903,4241,5552,90701   \n",
       "167                          1191,1509,2903,4241,10221   \n",
       "168                                    1191,5552,10257   \n",
       "\n",
       "                                               Symbols  \n",
       "0    PRDM1,CD27,FANCA,ITGB1,SPN,FZD7,TNFRSF13B,CADM...  \n",
       "1    PRDM1,CD27,FANCA,ITGB1,SPN,FZD7,TNFRSF13B,CADM...  \n",
       "2                     PRDM1,CD27,FANCA,SPN,FZD7,SEMA4A  \n",
       "3               PRDM1,CD27,FANCA,ITGB1,SPN,FZD7,SEMA4A  \n",
       "4         PRDM1,CD27,FANCA,ITGB1,SPN,FZD7,TRIB1,SEMA4A  \n",
       "..                                                 ...  \n",
       "164                               RUNX2,PBX1,SRGN,RORB  \n",
       "165      CLU,GRIN2A,MELTF,SRGN,SEC11C,CTSD,TRIB1,ABCC4  \n",
       "166                       CLU,GRIN2A,MELTF,SRGN,SEC11C  \n",
       "167                        CLU,CTSD,GRIN2A,MELTF,TRIB1  \n",
       "168                                     CLU,SRGN,ABCC4  \n",
       "\n",
       "[169 rows x 9 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_12 = pd.read_excel('pnas.2111920118.sd12.xlsx')\n",
    "data_12"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cytokines</th>\n",
       "      <th>HC Mean+/-SD (pg/mL)</th>\n",
       "      <th>MS Mean+/-SD (pg/mL)</th>\n",
       "      <th>P-value (MS vs HC)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TRAIL</td>\n",
       "      <td>76.63+/-25.83</td>\n",
       "      <td>82.18+/-27.87</td>\n",
       "      <td>0.5250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BAFF</td>\n",
       "      <td>259.66+/-40.73</td>\n",
       "      <td>241.36+/-47.18</td>\n",
       "      <td>0.3424</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IL6</td>\n",
       "      <td>21.92+/-70.82</td>\n",
       "      <td>1.66+/-3.52</td>\n",
       "      <td>0.7113</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>IL10</td>\n",
       "      <td>0.82+/-0.57</td>\n",
       "      <td>2.11+/-4.81</td>\n",
       "      <td>0.4027</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Cytokines HC Mean+/-SD (pg/mL) MS Mean+/-SD (pg/mL)  P-value (MS vs HC)\n",
       "0     TRAIL        76.63+/-25.83        82.18+/-27.87              0.5250\n",
       "1      BAFF       259.66+/-40.73       241.36+/-47.18              0.3424\n",
       "2       IL6        21.92+/-70.82          1.66+/-3.52              0.7113\n",
       "3      IL10          0.82+/-0.57          2.11+/-4.81              0.4027"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_13 = pd.read_excel('pnas.2111920118.sd13.xlsx')\n",
    "data_13"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Name</th>\n",
       "      <th>Sequence (5'&gt;3')</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>pSEMA4A for</td>\n",
       "      <td>GCCCCAGACTCATGGTGCCT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>pSEMA4A rev</td>\n",
       "      <td>GTCAGCTGGGGCCCATTCCT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>pPRDM1 for</td>\n",
       "      <td>GCTTTGATGAAGCCACATGT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>pPRDM1 rev</td>\n",
       "      <td>ACTTCAGTTCCGATTATGCT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>pTNFRSF17 for</td>\n",
       "      <td>ACTGTGGATTTTTAAAATGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>pTNFRSF17 rev</td>\n",
       "      <td>GAACTTCGCCTGCTTCGTGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>pDCAF12 for</td>\n",
       "      <td>GAGCACCTGTAATCTGAGCT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>pDCAF12 rev</td>\n",
       "      <td>TGACCTGCCATCTAAACCAC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>pZWINT for</td>\n",
       "      <td>TGAGACGGAGTCTCGCACTG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>pZWINT rev</td>\n",
       "      <td>CCTAAGATTACTTTGAAGTC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               Name      Sequence (5'>3')\n",
       "0  pSEMA4A for       GCCCCAGACTCATGGTGCCT\n",
       "1   pSEMA4A rev      GTCAGCTGGGGCCCATTCCT\n",
       "2  pPRDM1 for        GCTTTGATGAAGCCACATGT\n",
       "3        pPRDM1 rev  ACTTCAGTTCCGATTATGCT\n",
       "4     pTNFRSF17 for  ACTGTGGATTTTTAAAATGG\n",
       "5     pTNFRSF17 rev  GAACTTCGCCTGCTTCGTGG\n",
       "6       pDCAF12 for  GAGCACCTGTAATCTGAGCT\n",
       "7       pDCAF12 rev  TGACCTGCCATCTAAACCAC\n",
       "8        pZWINT for  TGAGACGGAGTCTCGCACTG\n",
       "9        pZWINT rev  CCTAAGATTACTTTGAAGTC"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_14 = pd.read_excel('pnas.2111920118.sd14.xlsx')\n",
    "data_14"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
