{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## GSE239626"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "| GEO Accession | Provided Information | Technology Used | Type of Data | Tissue/Cell Type Analyzed | Specific Samples |\n",
    "|---------------|----------------------|------------------|--------------|----------------------------|------------------|\n",
    "| GSE239626     | Gene expression profiling using high-throughput sequencing (scRNA-seq and CITE-seq) | scRNA-seq, CITE-seq | Processed matrix (gene expression) | Peripheral blood mononuclear cells (PBMCs) | Multiple sclerosis patients treated with high-dose vitamin D and placebo |"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "c:\\Users\\giuli\\anaconda3\\lib\\site-packages\\pandas\\core\\arrays\\masked.py:60: UserWarning: Pandas requires version '1.3.6' or newer of 'bottleneck' (version '1.3.5' currently installed).\n",
      "  from pandas.core import (\n"
     ]
    }
   ],
   "source": [
    "from GSE239626.load_GSE239626 import load_GSE239626\n",
    "\n",
    "genomic_matrix_df, gene_expression_df, antibody_capture_df, family_phenotype_df = load_GSE239626()\n",
    "# 7 min to load first time, 10 sec for subsequent runs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Help on function load_GSE239626 in module GSE239626.load_GSE239626:\n",
      "\n",
      "load_GSE239626()\n",
      "    Load data from GSE239626 dataset\n",
      "    \n",
      "    Returns:\n",
      "        genomic_matrix_df (pd.DataFrame): Genomic matrix data\n",
      "        gene_expression_df (pd.DataFrame): Gene expression data, subset of genomic matrix\n",
      "        antibody_capture_df (pd.DataFrame): Antibody capture data, subset of genomic matrix\n",
      "        family_phenotype_df (pd.DataFrame): Family phenotype data\n",
      "\n"
     ]
    }
   ],
   "source": [
    "help(load_GSE239626)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>symbol</th>\n",
       "      <th>expression</th>\n",
       "      <th>AAACCCAAGACTGTTC-1</th>\n",
       "      <th>AAACCCAAGGATCATA-1</th>\n",
       "      <th>AAACCCAGTTATTCTC-1</th>\n",
       "      <th>AAACCCATCATGAGGG-1</th>\n",
       "      <th>AAACGAAAGCCAGTAG-1</th>\n",
       "      <th>AAACGAAAGCTGACAG-1</th>\n",
       "      <th>AAACGAAAGTAATACG-1</th>\n",
       "      <th>...</th>\n",
       "      <th>TTTGGTTAGCTTTCCC-20</th>\n",
       "      <th>TTTGGTTGTTTCGACA-20</th>\n",
       "      <th>TTTGGTTTCAGACCCG-20</th>\n",
       "      <th>TTTGGTTTCTCATAGG-20</th>\n",
       "      <th>TTTGGTTTCTGTAACG-20</th>\n",
       "      <th>TTTGTTGAGCGTCGAA-20</th>\n",
       "      <th>TTTGTTGAGTGTACCT-20</th>\n",
       "      <th>TTTGTTGCAATTGCAC-20</th>\n",
       "      <th>TTTGTTGCATTCAGCA-20</th>\n",
       "      <th>TTTGTTGGTGGGTTGA-20</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000243485</td>\n",
       "      <td>MIR1302-2HG</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000237613</td>\n",
       "      <td>FAM138A</td>\n",
       "      <td>Gene Expression</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 72320 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                ID       symbol       expression  AAACCCAAGACTGTTC-1  \\\n",
       "0  ENSG00000243485  MIR1302-2HG  Gene Expression                   0   \n",
       "1  ENSG00000237613      FAM138A  Gene Expression                   0   \n",
       "\n",
       "   AAACCCAAGGATCATA-1  AAACCCAGTTATTCTC-1  AAACCCATCATGAGGG-1  \\\n",
       "0                   0                   0                   0   \n",
       "1                   0                   0                   0   \n",
       "\n",
       "   AAACGAAAGCCAGTAG-1  AAACGAAAGCTGACAG-1  AAACGAAAGTAATACG-1  ...  \\\n",
       "0                   0                   0                   0  ...   \n",
       "1                   0                   0                   0  ...   \n",
       "\n",
       "   TTTGGTTAGCTTTCCC-20  TTTGGTTGTTTCGACA-20  TTTGGTTTCAGACCCG-20  \\\n",
       "0                    0                    0                    0   \n",
       "1                    0                    0                    0   \n",
       "\n",
       "   TTTGGTTTCTCATAGG-20  TTTGGTTTCTGTAACG-20  TTTGTTGAGCGTCGAA-20  \\\n",
       "0                    0                    0                    0   \n",
       "1                    0                    0                    0   \n",
       "\n",
       "   TTTGTTGAGTGTACCT-20  TTTGTTGCAATTGCAC-20  TTTGTTGCATTCAGCA-20  \\\n",
       "0                    0                    0                    0   \n",
       "1                    0                    0                    0   \n",
       "\n",
       "   TTTGTTGGTGGGTTGA-20  \n",
       "0                    0  \n",
       "1                    0  \n",
       "\n",
       "[2 rows x 72320 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_expression_df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(36601, 72320)"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_expression_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "36601"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(gene_expression_df['ID'].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'NKG7' in gene_expression_df['symbol'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>symbol</th>\n",
       "      <th>expression</th>\n",
       "      <th>AAACCCAAGACTGTTC-1</th>\n",
       "      <th>AAACCCAAGGATCATA-1</th>\n",
       "      <th>AAACCCAGTTATTCTC-1</th>\n",
       "      <th>AAACCCATCATGAGGG-1</th>\n",
       "      <th>AAACGAAAGCCAGTAG-1</th>\n",
       "      <th>AAACGAAAGCTGACAG-1</th>\n",
       "      <th>AAACGAAAGTAATACG-1</th>\n",
       "      <th>...</th>\n",
       "      <th>TTTGGTTAGCTTTCCC-20</th>\n",
       "      <th>TTTGGTTGTTTCGACA-20</th>\n",
       "      <th>TTTGGTTTCAGACCCG-20</th>\n",
       "      <th>TTTGGTTTCTCATAGG-20</th>\n",
       "      <th>TTTGGTTTCTGTAACG-20</th>\n",
       "      <th>TTTGTTGAGCGTCGAA-20</th>\n",
       "      <th>TTTGTTGAGTGTACCT-20</th>\n",
       "      <th>TTTGTTGCAATTGCAC-20</th>\n",
       "      <th>TTTGTTGCATTCAGCA-20</th>\n",
       "      <th>TTTGTTGGTGGGTTGA-20</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>36601</th>\n",
       "      <td>CCR4</td>\n",
       "      <td>CCR4_TotalSeqB</td>\n",
       "      <td>Antibody Capture</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36602</th>\n",
       "      <td>CCR5</td>\n",
       "      <td>CCR5_TotalSeqB</td>\n",
       "      <td>Antibody Capture</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 72320 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         ID          symbol        expression  AAACCCAAGACTGTTC-1  \\\n",
       "36601  CCR4  CCR4_TotalSeqB  Antibody Capture                   5   \n",
       "36602  CCR5  CCR5_TotalSeqB  Antibody Capture                   0   \n",
       "\n",
       "       AAACCCAAGGATCATA-1  AAACCCAGTTATTCTC-1  AAACCCATCATGAGGG-1  \\\n",
       "36601                   0                   0                   1   \n",
       "36602                   1                   0                   0   \n",
       "\n",
       "       AAACGAAAGCCAGTAG-1  AAACGAAAGCTGACAG-1  AAACGAAAGTAATACG-1  ...  \\\n",
       "36601                   0                   0                   0  ...   \n",
       "36602                   0                   1                   0  ...   \n",
       "\n",
       "       TTTGGTTAGCTTTCCC-20  TTTGGTTGTTTCGACA-20  TTTGGTTTCAGACCCG-20  \\\n",
       "36601                    0                    2                    0   \n",
       "36602                    0                    0                    0   \n",
       "\n",
       "       TTTGGTTTCTCATAGG-20  TTTGGTTTCTGTAACG-20  TTTGTTGAGCGTCGAA-20  \\\n",
       "36601                    0                    0                    0   \n",
       "36602                    0                    0                    0   \n",
       "\n",
       "       TTTGTTGAGTGTACCT-20  TTTGTTGCAATTGCAC-20  TTTGTTGCATTCAGCA-20  \\\n",
       "36601                    0                    0                    3   \n",
       "36602                    0                    0                    0   \n",
       "\n",
       "       TTTGTTGGTGGGTTGA-20  \n",
       "36601                    0  \n",
       "36602                    3  \n",
       "\n",
       "[2 rows x 72320 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "antibody_capture_df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(35, 72320)"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "antibody_capture_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['CCR4', 'CCR5', 'CCR6', 'CCR7', 'CD11a', 'CD127', 'CD138', 'CD161',\n",
       "       'CD20', 'CD226', 'CD24', 'CD25', 'CD27', 'CD3', 'CD38', 'CD4',\n",
       "       'CD45RA', 'CD49b', 'CD49d', 'CD56', 'CD62L', 'CD71', 'CD8',\n",
       "       'CTLA-4', 'CXCR3', 'CXCR4', 'CXCR5', 'GITR', 'HLA-DR', 'ICOS',\n",
       "       'IgD', 'IgM', 'LAG3', 'TCRVa7.2', 'TCRgd'], dtype=object)"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "antibody_capture_df['ID'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>geo_accession</th>\n",
       "      <th>status</th>\n",
       "      <th>submission_date</th>\n",
       "      <th>last_update_date</th>\n",
       "      <th>type</th>\n",
       "      <th>channel_count</th>\n",
       "      <th>source_name_ch1</th>\n",
       "      <th>organism_ch1</th>\n",
       "      <th>taxid_ch1</th>\n",
       "      <th>...</th>\n",
       "      <th>contact_zip/postal_code</th>\n",
       "      <th>contact_country</th>\n",
       "      <th>instrument_model</th>\n",
       "      <th>library_selection</th>\n",
       "      <th>library_source</th>\n",
       "      <th>library_strategy</th>\n",
       "      <th>relation</th>\n",
       "      <th>supplementary_file_1</th>\n",
       "      <th>series_id</th>\n",
       "      <th>data_row_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GSM7669046</th>\n",
       "      <td>Patient Placebo N1 J0, GEX</td>\n",
       "      <td>GSM7669046</td>\n",
       "      <td>Public on Oct 05 2023</td>\n",
       "      <td>Jul 31 2023</td>\n",
       "      <td>Oct 05 2023</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>9606</td>\n",
       "      <td>...</td>\n",
       "      <td>34094</td>\n",
       "      <td>France</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>transcriptomic single cell</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>BioSample: https://www.ncbi.nlm.nih.gov/biosam...</td>\n",
       "      <td>NONE</td>\n",
       "      <td>GSE239626</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM7669047</th>\n",
       "      <td>Patient Placebo N1 J0, ADT</td>\n",
       "      <td>GSM7669047</td>\n",
       "      <td>Public on Oct 05 2023</td>\n",
       "      <td>Jul 31 2023</td>\n",
       "      <td>Oct 05 2023</td>\n",
       "      <td>SRA</td>\n",
       "      <td>1</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>9606</td>\n",
       "      <td>...</td>\n",
       "      <td>34094</td>\n",
       "      <td>France</td>\n",
       "      <td>Illumina NovaSeq 6000</td>\n",
       "      <td>cDNA</td>\n",
       "      <td>other</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>BioSample: https://www.ncbi.nlm.nih.gov/biosam...</td>\n",
       "      <td>NONE</td>\n",
       "      <td>GSE239626</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 title geo_accession                 status  \\\n",
       "GSM7669046  Patient Placebo N1 J0, GEX    GSM7669046  Public on Oct 05 2023   \n",
       "GSM7669047  Patient Placebo N1 J0, ADT    GSM7669047  Public on Oct 05 2023   \n",
       "\n",
       "           submission_date last_update_date type channel_count  \\\n",
       "GSM7669046     Jul 31 2023      Oct 05 2023  SRA             1   \n",
       "GSM7669047     Jul 31 2023      Oct 05 2023  SRA             1   \n",
       "\n",
       "           source_name_ch1  organism_ch1 taxid_ch1  ...  \\\n",
       "GSM7669046           Blood  Homo sapiens      9606  ...   \n",
       "GSM7669047           Blood  Homo sapiens      9606  ...   \n",
       "\n",
       "           contact_zip/postal_code contact_country       instrument_model  \\\n",
       "GSM7669046                   34094          France  Illumina NovaSeq 6000   \n",
       "GSM7669047                   34094          France  Illumina NovaSeq 6000   \n",
       "\n",
       "           library_selection              library_source library_strategy  \\\n",
       "GSM7669046              cDNA  transcriptomic single cell          RNA-Seq   \n",
       "GSM7669047              cDNA                       other          RNA-Seq   \n",
       "\n",
       "                                                     relation  \\\n",
       "GSM7669046  BioSample: https://www.ncbi.nlm.nih.gov/biosam...   \n",
       "GSM7669047  BioSample: https://www.ncbi.nlm.nih.gov/biosam...   \n",
       "\n",
       "           supplementary_file_1  series_id data_row_count  \n",
       "GSM7669046                 NONE  GSE239626              0  \n",
       "GSM7669047                 NONE  GSE239626              0  \n",
       "\n",
       "[2 rows x 35 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "family_phenotype_df.head(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## GSE159033"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "| GEO Accession | Provided Information | Technology Used | Type of Data | Tissue/Cell Type Analyzed | Specific Samples |\n",
    "|---------------|----------------------|------------------|--------------|----------------------------|------------------|\n",
    "| GSE159033     | Whole-transcriptome profiling in PBMCs from patients with LS-OCMB characterization reveals 4 RNAs as biomarkers of highly active disease | Non-coding RNA profiling by array, RNA-seq | Processed data (miRNA) | Peripheral blood mononuclear cells (PBMCs) | Multiple sclerosis patients with positive and negative LS-OCMB status |\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "from GSE159033.load_GSE159033 import load_GSE159033\n",
    "\n",
    "genomic_matrix_df, family_phenotype_df = load_GSE159033()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Help on function load_GSE159033 in module GSE159033.load_GSE159033:\n",
      "\n",
      "load_GSE159033()\n",
      "    Load data from GSE159033 dataset\n",
      "    \n",
      "    Returns:\n",
      "        genomic_matrix_df (pd.DataFrame): Genomic matrix data\n",
      "        family_phenotype_df (pd.DataFrame): Family phenotype data\n",
      "\n"
     ]
    }
   ],
   "source": [
    "help(load_GSE159033)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Accession_ID</th>\n",
       "      <th>Accession</th>\n",
       "      <th>Transcript ID(Array Design)</th>\n",
       "      <th>Sequence Type</th>\n",
       "      <th>Species Scientific Name</th>\n",
       "      <th>Alignments</th>\n",
       "      <th>Sequence Length</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>Genome Context</th>\n",
       "      <th>...</th>\n",
       "      <th>GSM4817990</th>\n",
       "      <th>GSM4817991</th>\n",
       "      <th>GSM4817992</th>\n",
       "      <th>GSM4817993</th>\n",
       "      <th>GSM4817994</th>\n",
       "      <th>GSM4817995</th>\n",
       "      <th>GSM4817996</th>\n",
       "      <th>GSM4817997</th>\n",
       "      <th>GSM4817998</th>\n",
       "      <th>GSM4817999</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20500112</td>\n",
       "      <td>MIMAT0000062_st</td>\n",
       "      <td>MIMAT0000062</td>\n",
       "      <td>hsa-let-7a-5p</td>\n",
       "      <td>miRNA</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>chr11:122017276-122017297 (-) /// chr22:465086...</td>\n",
       "      <td>22</td>\n",
       "      <td>UGAGGUAGUAGGUUGUAUAGUU</td>\n",
       "      <td>ENST00000362295 // sense // exon // 1 /// ENST...</td>\n",
       "      <td>...</td>\n",
       "      <td>10.98</td>\n",
       "      <td>12.64</td>\n",
       "      <td>11.17</td>\n",
       "      <td>11.89</td>\n",
       "      <td>11.50</td>\n",
       "      <td>11.56</td>\n",
       "      <td>11.73</td>\n",
       "      <td>11.55</td>\n",
       "      <td>12.16</td>\n",
       "      <td>11.64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20500113</td>\n",
       "      <td>MIMAT0004481_st</td>\n",
       "      <td>MIMAT0004481</td>\n",
       "      <td>hsa-let-7a-3p</td>\n",
       "      <td>miRNA</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>chr22:46508680-46508700 (+) /// chr9:96938295-...</td>\n",
       "      <td>21</td>\n",
       "      <td>CUAUACAAUCUACUGUCUUUC</td>\n",
       "      <td>ENST00000362295 // sense // exon // 1 /// ENST...</td>\n",
       "      <td>...</td>\n",
       "      <td>1.34</td>\n",
       "      <td>0.89</td>\n",
       "      <td>1.11</td>\n",
       "      <td>1.14</td>\n",
       "      <td>0.77</td>\n",
       "      <td>0.96</td>\n",
       "      <td>0.79</td>\n",
       "      <td>1.21</td>\n",
       "      <td>1.35</td>\n",
       "      <td>1.38</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         ID     Accession_ID     Accession Transcript ID(Array Design)  \\\n",
       "0  20500112  MIMAT0000062_st  MIMAT0000062               hsa-let-7a-5p   \n",
       "1  20500113  MIMAT0004481_st  MIMAT0004481               hsa-let-7a-3p   \n",
       "\n",
       "  Sequence Type Species Scientific Name  \\\n",
       "0         miRNA            Homo sapiens   \n",
       "1         miRNA            Homo sapiens   \n",
       "\n",
       "                                          Alignments  Sequence Length  \\\n",
       "0  chr11:122017276-122017297 (-) /// chr22:465086...               22   \n",
       "1  chr22:46508680-46508700 (+) /// chr9:96938295-...               21   \n",
       "\n",
       "                 Sequence                                     Genome Context  \\\n",
       "0  UGAGGUAGUAGGUUGUAUAGUU  ENST00000362295 // sense // exon // 1 /// ENST...   \n",
       "1   CUAUACAAUCUACUGUCUUUC  ENST00000362295 // sense // exon // 1 /// ENST...   \n",
       "\n",
       "   ... GSM4817990 GSM4817991 GSM4817992 GSM4817993 GSM4817994 GSM4817995  \\\n",
       "0  ...      10.98      12.64      11.17      11.89      11.50      11.56   \n",
       "1  ...       1.34       0.89       1.11       1.14       0.77       0.96   \n",
       "\n",
       "  GSM4817996  GSM4817997  GSM4817998  GSM4817999  \n",
       "0      11.73       11.55       12.16       11.64  \n",
       "1       0.79        1.21        1.35        1.38  \n",
       "\n",
       "[2 rows x 29 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genomic_matrix_df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>title</th>\n",
       "      <th>geo_accession</th>\n",
       "      <th>status</th>\n",
       "      <th>submission_date</th>\n",
       "      <th>last_update_date</th>\n",
       "      <th>type</th>\n",
       "      <th>channel_count</th>\n",
       "      <th>source_name_ch1</th>\n",
       "      <th>organism_ch1</th>\n",
       "      <th>taxid_ch1</th>\n",
       "      <th>...</th>\n",
       "      <th>contact_name</th>\n",
       "      <th>contact_email</th>\n",
       "      <th>contact_institute</th>\n",
       "      <th>contact_address</th>\n",
       "      <th>contact_city</th>\n",
       "      <th>contact_zip/postal_code</th>\n",
       "      <th>contact_country</th>\n",
       "      <th>supplementary_file</th>\n",
       "      <th>series_id</th>\n",
       "      <th>data_row_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GSM4817989</th>\n",
       "      <td>Sample04-positive</td>\n",
       "      <td>GSM4817989</td>\n",
       "      <td>Public on Dec 21 2020</td>\n",
       "      <td>Oct 05 2020</td>\n",
       "      <td>Dec 22 2020</td>\n",
       "      <td>RNA</td>\n",
       "      <td>1</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>9606</td>\n",
       "      <td>...</td>\n",
       "      <td>Maider,,MuÃ±oz-Culla</td>\n",
       "      <td>maider.munoz@ehu.eus</td>\n",
       "      <td>Biodonostia Institute</td>\n",
       "      <td>Paseo Dr.Begiristain s/n</td>\n",
       "      <td>San Sebastian</td>\n",
       "      <td>20014</td>\n",
       "      <td>Spain</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4817...</td>\n",
       "      <td>GSE159033,GSE159036</td>\n",
       "      <td>6631</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GSM4817990</th>\n",
       "      <td>Sample05-positive</td>\n",
       "      <td>GSM4817990</td>\n",
       "      <td>Public on Dec 21 2020</td>\n",
       "      <td>Oct 05 2020</td>\n",
       "      <td>Dec 22 2020</td>\n",
       "      <td>RNA</td>\n",
       "      <td>1</td>\n",
       "      <td>Blood</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>9606</td>\n",
       "      <td>...</td>\n",
       "      <td>Maider,,MuÃ±oz-Culla</td>\n",
       "      <td>maider.munoz@ehu.eus</td>\n",
       "      <td>Biodonostia Institute</td>\n",
       "      <td>Paseo Dr.Begiristain s/n</td>\n",
       "      <td>San Sebastian</td>\n",
       "      <td>20014</td>\n",
       "      <td>Spain</td>\n",
       "      <td>ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4817...</td>\n",
       "      <td>GSE159033,GSE159036</td>\n",
       "      <td>6631</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        title geo_accession                 status  \\\n",
       "GSM4817989  Sample04-positive    GSM4817989  Public on Dec 21 2020   \n",
       "GSM4817990  Sample05-positive    GSM4817990  Public on Dec 21 2020   \n",
       "\n",
       "           submission_date last_update_date type  channel_count  \\\n",
       "GSM4817989     Oct 05 2020      Dec 22 2020  RNA              1   \n",
       "GSM4817990     Oct 05 2020      Dec 22 2020  RNA              1   \n",
       "\n",
       "           source_name_ch1  organism_ch1  taxid_ch1  ...  \\\n",
       "GSM4817989           Blood  Homo sapiens       9606  ...   \n",
       "GSM4817990           Blood  Homo sapiens       9606  ...   \n",
       "\n",
       "                    contact_name         contact_email      contact_institute  \\\n",
       "GSM4817989  Maider,,MuÃ±oz-Culla  maider.munoz@ehu.eus  Biodonostia Institute   \n",
       "GSM4817990  Maider,,MuÃ±oz-Culla  maider.munoz@ehu.eus  Biodonostia Institute   \n",
       "\n",
       "                     contact_address   contact_city contact_zip/postal_code  \\\n",
       "GSM4817989  Paseo Dr.Begiristain s/n  San Sebastian                   20014   \n",
       "GSM4817990  Paseo Dr.Begiristain s/n  San Sebastian                   20014   \n",
       "\n",
       "           contact_country                                 supplementary_file  \\\n",
       "GSM4817989           Spain  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4817...   \n",
       "GSM4817990           Spain  ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4817...   \n",
       "\n",
       "                      series_id data_row_count  \n",
       "GSM4817989  GSE159033,GSE159036           6631  \n",
       "GSM4817990  GSE159033,GSE159036           6631  \n",
       "\n",
       "[2 rows x 32 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "family_phenotype_df.head(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##  GSE173787\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### First dataset\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Contains 12 HC (healthy condition) patients and 17 MS (multiple sclerosis) patients."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
